---
title: "An investigation into the contentious phylogenetic results of the *Mixta* species"
output:
  word_document:
    reference_docx: Style_Manuscript.docx
    fig_caption: true
    # toc: true
    # toc_depth: 5
bibliography: References.bib
csl: IJSEM.csl
---
```{r Packages, echo = FALSE, message = FALSE, warning=FALSE}
library("knitr")
library("tidyverse")
library("ggplot2")
library("ggpubr")
library("RColorBrewer")
library("extrafont")
theme_set(theme_classic())

Spp <- c("P_septica", "P_agglomerans", "E_tasmaniensis", "E_amylovora", "T_ptyseos", "T_saanichensis", 
         "E_cloacae", "P_syringae")
```

Kimberly N. Hinz, John Stavrinides^1^

Department of Biology  
University of Regina  
3737 Wascana Parkway  
Regina, Saskatchewan, Canada  
S4S0A2

^1^**Correspondence**:  
John Stavrinides  
Department of Biology  
University of Regina  
3737 Wascana Parkway  
Regina, Saskatchewan  
S4S0A2  
Ph: (306) 337-8478  
Email: john.stavrinides@uregina.ca

**Keywords:**

<!-- \newpage -->
# Abstract
<!-- *Mixta* is a recently described genus within the bacterial family *Erwiniaceae* and is closely related to *Pantoea*, *Erwinia*, and *Tatumella*. Depending on the genes and method used for phylogenetic analysis, the genus with which *Mixta* shares the most recent ancestor differs. This study aimed to determine the cause behind these contentious results and identified genes that may have resulted from horizontal gene transfer. Ten complete or partial genomes – two *Mixta* species, six species from other *Erwiniaceae* genera, and two outgroups – were retrieved from NCBI and annotated. Homologous genes were extracted yielding a dataset of 799 genes. The genes were aligned with the ClustalW algorithm, and MEGA-CC was used for model testing and computing phylogenetic trees for each gene for both nucleotide and amino acid sequences. Distance matrices were then extracted from these trees. Nucleotide and amino acid identity analyses were also done using the programming language `R`. *Pantoea* was the closest relative to the *Mixta* species in most analyses; however, results were not consistent. Some genes were also found to be more similar to other, non-*Pantoea* species. Diligence must be given to selecting genes for phylogenetic analysis and the method chosen to prevent any xenologous signal from distorting the actual relationships. Furthermore, future research should consider that different phylogenetic analyses may provide different results. -->

# Intro and MM
<!-- \newpage -->
<!-- # Introduction -->
<!-- The bacterial family *Erwiniaceae* includes diverse Gram-negative [@Winslow1917; @Prakash2015] organisms, including the genera *Mixta*, *Pantoea*, *Erwinia*, *Tatumella*, *Buchnera*, *Wigglesworthia*, *Phaseolibacter*, and *Kalamiella* [REF CRAIG]. *Mixta* is a relatively newer genus first identified by Palmer et al. in 2017 [@Palmer2017] and later described by the same researchers in 2018 [@Palmer2018]. Currently, there are only five *Mixta* species validly published [@LPSN]. *M. calida* and *M. gaviniae* (previously species of *Pantoea*) were initially found in infant formula and its production environment [@Popp2010]. Fritz et al. [@Fritz2014] demonstrated that some strains of *M. calida* were linked with post-surgical meningitis in humans. The other *Mixta* species include *M. theicola* which was isolated from black tea extract [@Kato2015], *M. intestinalis* which was isolated from a healthy human fecal sample [@Prakash2015], and *M. tenebrionis* which was isolated from the gut microbiota of a plastic-eating mealworm [@Xia2020]. The first two were previously named as *Pantoea* species. -->

<!-- *Mixta*'s position relative to other members of the *Erwiniaceae*, in particular to *Pantoea*, *Erwinia*, *Tatumella*, differs vastly between studies [@Prakash2015; @Palmer2017; @Palmer2018; @Adeolu2016; @Brady2011; @Rezzonico2016; @Brady2012; @Gueule2015]. Prakash et al. [@Prakash2015], Palmer et al. [@Palmer2017], Rezzonico et al. [@Rezzonico2016], and Gueule et al. [@Gueule2015] all used *gyrB*, *rpoB*, *atpD*, and *infB* nucleotide gene sequences in their phylogenetic analyses. Prakash et al. [@Prakash2015] conducted a neighbour-joining statistical method while the others used a maximum-likelihood (ML) method [@Palmer2017; @Rezzonico2016; @Gueule2015]. Three of the studies – Rezzonico et al. [@Rezzonico2016], Prakash et al. [@Prakash2015], and Gueule et al. [@Gueule2015] – placed the *Mixta* species within the *Pantoea* genera. In contrast, the analysis in Palmer et al. [@Palmer2017] identified *Mixta* outside of the other three genera. A separate study by Palmer et al. [@Palmer2018], Rezzonico et al. [@Rezzonico2016], and Brady et al. [@Brady2012] also ran analyses with the four MLSA genes as amino acid sequences. *Mixta* was found to be related closest to *Tatumella* by Rezzonico et al. [@Rezzonico2016] and *Pantoea* by Palmer et al. [@Palmer2018]. Brady et al. [@Brady2012] suggested that *Mixta* was a part of the *Pantoea* genus and not separate. Additionally, Palmer et al. [@Palmer2017] computed two ML trees; one used 1039 core amino acid sequences, and the other used 52 ribosomal MLSA nucleotide sequences. The 1039 core gene tree suggested *Pantoea* and *Tatumella* were sister taxa followed by two *Mixta* species. However, the ribosomal MLSA gene tree proposed *Pantoea* and *Tatumella* were sister taxa followed by *Erwinia* and then by *Mixta*. -->

<!-- What remains unclear is the relative contribution of disaparate evolutionary histories of *Mixta* genes, tree-building methods, and sequence evolution on the phylogenetic position of the genus. This paper aims to determine the cause behind the *Mixta* species' contentious phylogenetic results. Ten complete or partial genomes – two for each of *Mixta*, *Pantoea*, *Erwinia*, and *Tatumella* and two outgroups – were retrieved from a public database. Homologous genes were extracted from these genomes in both the nucleotide and amino acid sequences. Distance matrices were exported from the phylogenetic trees, and these were used to determine which species were most closely related to *Mixta* for each gene. Nucleotide and amino acid identity analyses were also conducted for each gene to compare the results between evolutionary history and sequence similarity. Phylogenetic trees were constructed for combinations of different gene sequences.  -->

<!-- # Materials and Methods -->
<!-- ## Genomes and Gene Sets -->
<!-- Ten publicly available full or partial genomes of the species' type strains were retrieved from the National Center for Biotechnology Information (NCBI). Two representative genomes were used for each of the four *Erwiniaceae* genera while *Enterobacter cloacae* (*Enterobacterales*) and *Pseudomonas syringae* (*Pseudomonaceae*) were used as outgroups (Table 1). The genomes were annotated with Prokka v1.14.1 [@Seemann2014] using default parameters, and homologous genes were extracted using GET_HOMOLOGUES software package [@Contreras-Moreira2013] with the bidirectional best-hit search algorithm using default parameters. GET_HOMOLOGUES extracted 954 homologous genes between the ten species. -->

<!-- ```{r Table1, echo=FALSE, message=FALSE, warning=FALSE, fig.cap=TRUE} -->
<!-- strain <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Table1_Genomes.csv", stringsAsFactors = FALSE) -->

<!-- colnames(strain)[3] <- "GenBank Accession No." -->

<!-- kable(strain, caption = "Table 1. Genomes used in this study.") -->
<!-- ``` -->

<!-- In `R` [@RCore2019], two filter functions were written for the nucleotide sequence gene files. The first filter kept genes files that contained ten, single-copy sequences that were approximately the same length. The shortest gene length allowed was 90% the length of the longest sequence in the gene file. Of the 954 homologous gene sets, 38 multi-copy genes were excluded, and another 175 were excluded for not meeting the length requirements.  -->

<!-- A second filter was applied to these 175 genes to identify any genes wherein both representatives of each genera were approximately the same length. The filter split the genes into two groups: sequences that met the original 90% length cutoff and sequences that were between 80 and 90% the length of the longest sequence. If both representatives of a genus were in the same group, then the gene was added to the working dataset. If at least one of the sequences did not meet the 80% cutoff, then the gene was excluded. An additional 58 genes were added to the working dataset, resulting in 799 genes. The amino acid sequences for these genes were subsequently downloaded. -->

<!-- ## Gene Alignment -->
<!-- Gene alignments were done in `R` using a ClustalW algorithm. The fasta files were read into the session using the `readDNAStringSet()` function (package: Biostrings) [@Pages2020]. The function called to do the alignments was the `msaClustalW()` function (package: msa) [@Bodenhofer2015] using default parameters and 100 iterations. The aligned sequences were then converted into a comma-separated values (CSV)-compatible format using `msaConvert()` (package: msa)[@Bodenhofer2015] and written into a new fasta file using the `write.fasta()` function (package: seqinr) [@Charif2007] -->

<!-- ## Model Selection -->
<!-- The best model of sequence evolution was identified for each nucleotide and amino acid alignment using Model Selection as implemented in MEGA-CC [@Kumar2012]. Default parameters were used for model testing. The output was a CSV file for each gene. These CSV files were read into `R`, and the model for phylogenetic analysis was extracted based on the lowest Bayesian Information Criterion (BIC). -->

<!-- ## Phylogenetic Analysis -->
<!-- Phylogenetic analysis was performed in MEGA-CC using maximum likelihood with 500 bootstrap replications, the model, and rate patterns to create a phylogenetic tree for each gene. If a model required a Gamma distribution, the number of discrete Gamma categories was set to 5, which is the default. -->

<!-- Distance matrices were extracted from each of the phylogenetic trees and saved as Excel files. These files were then read into the `R` session. The genetic distances between *Mixta* and the other species were extracted for further analysis. Tables were created in `R Markdown` using the `kable()` function (package: knitr) [@Xie2020] and figures were created using the `ggplot()` function (package: ggplot2) [@Wickham2016]. -->

<!-- ## Nucleotide and Amino Acid Identity -->
<!-- Each of the non-*Mixta* sequences were aligned to one of the corresponding *Mixta* sequences in turn. Alignments were performed in R as described above, and the aligned sequences were then split into the individual sites using the `str_split()` function (package: stringr) [@Wickham2019]. A function was created using base `R` code to calculate the number of identical sites between each sequence and the *Mixta* homologs. Values that equal or are close to 100% are more identical to the *Mixta* sequences. -->

# Results
After the two filters, 799 of 954 homologous genes remained. The genes included those encoding rRNA sequences, enzymes, transporters, and housekeeping genes and were scattered across the M. calida and M. gaviniae genomes.

## Best Model vs Best Available Model
In MEGAX [@Kumar2018], the user can either calculate genetic distance or estimate a phylogenetic tree. However, although model testing assesses all possible models available in the software, all models are not available for genetic distance analysis. The GTR (General Time Reversible) and HKY (Hasegawa-Kishino-Yano) models and the +I (invariant sites) parameter are available only for phylogenetic tree analysis. 

The genetic distances for the nucleotide sequences (NTS) were estimated using both the recommended (RM) and best available models (BAM) for distance analysis to determine if different results would arise when using a model that does not explain most of the genetic variance. The RM is the model with the lowest Bayesian Information Criterion (BIC) among the available options. First, phylogenetic gene trees were computed using the recommended model from model testing. The software suggested twelve unique models for this method (Table 2). Genetic distance matrices were then extracted from the trees. Second, distance matrices were calculated directly using the BAM for calculating pairwise distances. Only four models were recommended for genetic distance analysis (Table 2). Of the 799 NTS, 242 genes required a different model between the two analyses, and the mean BIC difference between the two models was 13.8 (range: 0.1, 122.6). 

```{r Table2, echo=FALSE, message=FALSE, warning=FALSE, fig.cap=TRUE}
models <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Table2_BestAvailableModels.csv", stringsAsFactors = FALSE)

colnames(models)[2:3] <- c("Best Model", "Best Available Model")

kable(models, 
      caption = "Table 2. The number of genes that required each model according to the best model with the lowest BIC and the best available model for genetic distance analysis.")
```

The closest relatives to both of the *Mixta* species were extracted from RM and BAM distance matrices. When the RM was used. *P. septica* was the closest relative to 66.3 and 66.5% of the *M. calida* and *M. gaviniae* sequences, respectively (Figure 1). These percentages increase by 12.6 and 14.4% for the *M. calida* and *M. gaviniae* sequences, respectively, when the BAM was used (Figure 1). Instead of *P. septica* when the RM is used, more genes were more closely related to *P. agglomerans*, *E. tasmaniensis*, *E. amylovora*, *T. saanichensis*, and *E. cloacae* (Figure 1). Additionally, there was more disparity between *M. calida* and *M. gaviniae* when the BAM was used (Figure 1).

```{r Figure1, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="Figure 1. The percentage of genes (n = 799) that were most closely related to any of the eight non-*Mixta* species when using the recommended model from model testing versus the best available) model for genetic distance analysis in MEGAX. The sequences were nucleotide sequences. The species with the shortest genetic distance from the *Mixta* species is considered to be the closest relative. The pink and orange bars represent the percentage of *Mixta* genes that were most closely related to any of the non-*Mixta* species when using the recommended model and the best available model, respectively. The lighter shades represent *M. calida* and the darker shades represent *M. gaviniae*.", fig.width=6.5}
number_relatives <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure1_FirstRelativeBestvsAvailableModelNT.csv", 
                             stringsAsFactors = FALSE) %>%
  mutate(Species = factor(Species, levels = c("P. septica", "P. agglomerans", "E. tasmaniensis", "E. amylovora", "T. ptyseos", "T. saanichensis", 
                                              "E. cloacae", "P. syringae"), ordered = TRUE),
         Type = factor(Type, levels = c("PD_calida", "PD_gaviniae", "GD_calida", "GD_gaviniae"), ordered = TRUE))

ggplot(data = number_relatives, aes(x = Species, y = Percent, fill = Type)) +
  geom_bar(stat = "identity", position = position_dodge()) +
  scale_fill_manual(values = c("#fb9a99", "#e31a1c", "#fdbf6f", "#ff7f00"),
                    labels = c(expression(paste("Recommended Model and ", italic("M. calida"), sep = "")), 
                               expression(paste("Recommended Model and ", italic("M. gaviniae"), sep = "")), 
                               expression(paste("Best Available Model and ", italic("M. calida"), sep = "")), 
                               expression(paste("Best Available Model and ", italic("M. gaviniae"), sep = "")))) +
  labs(y = "Percent of Genes", 
       fill = expression(paste("Model and ", italic("Mixta"), " species", sep = ""))) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, face = "italic"),               # Rotates and italicizes x axis labels
        legend.position = c(0.95, 0.95),                                                  # Moves legend inside the plot and in the top-right corner
        legend.justification = c("right", "top"),                                         # Top right corner is at above coordinates
        legend.text.align = 0,                                                            # Aligns legend text to the left
        legend.margin = margin(6, 6, 6, 6),                                               # Margins around legend
        text = element_text(size = 12,  family = "Times New Roman"))
```

The RM and BAM were different for approximately 30.3% (n = 242) of the genes. Of these 242 genes, 20.2% (n = 49) were said to have different closest relatives between the two analyses. However, when the RM and BAM were the same (n = 557), 19.9% (n = 111) had different closest relatives.

## Closest Relative from Genetic Distance
Genetic distance gives a numerical estimate of evolutionary change between species, and the taxa with which a species of interest has the shortest distance is regarded as the closest relative. Since the phylogenetic trees were calculated using the RM, the distance matrices exported from the trees were kept for future analyses. *P. septica* was consistently the closest relative for most of the *Mixta* sequences (Figure 2). However, the number of genes most closely related *P. septica* decreased from 66.3 and 66.5% (*M. calida* and *M. gaviniae*, respectively) with the NTS by 23.7% with the amino acid sequences (AAS; Figure 2). Instead, the *Mixta* AAS were most closely related to the other taxa within the dataset except for *P. syringae* (Figure 2). Interestingly, the number of genes most closely related to *E. cloacae* increased between the NTS and the AAS from 2.9% by 3.1% (Figure 2). Additionally, 55.1 and 55.2% of the *M. calida* and *M. gaviniae* sequences, respectively, had the same closest relatives between the NTS and the AAS analyses.

```{r Figure2, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="Figure 2. The percentage of genes (n = 799) that were most closely related to any of the eight non-*Mixta* species using the recommended models and genetic distances. The species with the shortest genetic distance from the *Mixta* species is considered to be the closest relative. The percentage of *Mixta* genes that were most closely related to any of the non-*Mixta* species are represented by the blue and green bars for the nucleotide and amino acid sequences, respectively. The lighter shades represent *M. calida* and the darker shades represent *M. gaviniae*.", fig.width=6.5}
number_relatives <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure2_FirstRelativeNTAA.csv", 
                             stringsAsFactors = FALSE) %>%
  mutate(Species = factor(Species, levels = c("P. septica", "P. agglomerans", "E. tasmaniensis", "E. amylovora", "T. ptyseos", "T. saanichensis", 
                                              "E. cloacae", "P. syringae"), ordered = TRUE),
         Type = factor(Type, levels = c("calida_NT", "gaviniae_NT", "calida_AA", "gaviniae_AA"), ordered = TRUE))


ggplot(data = number_relatives, aes(x = Species, y = Percent, fill = Type)) +
  geom_bar(stat = "identity", position = position_dodge()) +
  scale_fill_brewer(palette = "Paired", 
                    labels = c(expression(paste("NT and ", italic("M. calida"), sep = "")), 
                               expression(paste("NT and ", italic("M. gaviniae"), sep = "")), 
                               expression(paste("AA and ", italic("M. calida"), sep = "")), 
                               expression(paste("AA and ", italic("M. gaviniae"), sep = "")))) +
  labs(y = "Percent of Genes", 
       fill = expression(paste("Sequence type and ", italic("Mixta"), " species", sep = ""))) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, face = "italic"),               # Rotates and italicizes x axis labels
        legend.position = c(0.95, 0.95),                                                  # Moves legend inside the plot and in the top-right corner
        legend.justification = c("right", "top"),                                         # Top right corner is at above coordinates
        legend.text.align = 0,                                                            # Aligns legend text to the left
        legend.margin = margin(6, 6, 6, 6),                                               # Margins around legend
        text = element_text(size = 12,  family = "Times New Roman"))
```

## Spatial Patterns in the Closest Relatives from Genetic Distances
Circular plots were generated using the *Mixta* gene IDs to simulate their relative positions on the *Mixta* chromosomes and uncover positional and spatial patterns. The bars’ lengths and colours correspond to the species that was the closest relative to the *Mixta* gene at that position. The results between the two *Mixta* species for the closest relatives were the same except for one (*ttuB*) and two (*ttuB* and *secY*) genes for the NTS and the AAS, respectively; therefore, only the results for *M. calida* are shown.

Many of the genes most closely related to any of the other taxa in this study were evenly positioned around the *Mixta* chromosome in the NTS (Figure 3A). The exception is for the *Tatumella* species (pinks) and *E. amylovora* (dark green; Figure 3A). The former and most of the genes closely related to the latter were located on the right side of the chromosome (Figure 3A). Unexpectedly, genes closely related to *E. cloacae* (orange) were distributed throughout the *Mixta* genome.

```{r Figure3, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="Figure 3. The closest relatives to *M. calida* genes for (**A**) nucleotide and (**B**) amino acid sequences around the *M. calida* chromosome. The length and colour of the bars represent the different species to which the gene is most closely related according to genetic distance. The *M. calida* genome has 4092 genes.", fig.width=6.5, fig.height=3.9}
FR_nucl_calida <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure3A_Circular_NT.csv", stringsAsFactors = FALSE) %>%
  mutate(Closest_Relative = factor(Closest_Relative, 
                                   levels = c("P_septica", "P_agglomerans", "E_tasmaniensis", "E_amylovora", "T_ptyseos", 
                                              "T_saanichensis", "E_cloacae", "P_syringae"), ordered = TRUE))

FR_aa_calida <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure3B_Circular_AA.csv", stringsAsFactors = FALSE) %>%
  mutate(Closest_Relative = factor(Closest_Relative, 
                                   levels = c("P_septica", "P_agglomerans", "E_tasmaniensis", "E_amylovora", "T_ptyseos", 
                                              "T_saanichensis", "E_cloacae", "P_syringae"), ordered = TRUE))

cal_NT <- ggplot(FR_nucl_calida, aes(x = ID, y = Relative_Number, fill = Closest_Relative)) +
  geom_bar(stat = "identity", position = position_dodge(), width = 20) +
  coord_polar() +
  scale_fill_brewer(palette = "Paired", 
                    labels = c("P. septica", "P. agglomerans", "E. tasmaniensis", "E. amylovora", "T. ptyseos", "T. saanichensis", "E. cloacae",
                               "P. syringae")) +
  scale_y_continuous(limits = c(0, 9), breaks = c(0, 2, 4, 6, 8)) +
  labs(fill = "Closest Relative") +
  theme_bw() +
  theme(legend.text = element_text(face = "italic"),
        axis.title = element_blank(),
        panel.border = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        text = element_text(size = 12,  family = "Times New Roman"))

cal_AA <- ggplot(FR_aa_calida, aes(x = ID, y = Relative_Number, fill = Closest_Relative)) +
  geom_bar(stat = "identity", position = position_dodge(), width = 20) +
  coord_polar() +
  scale_fill_brewer(palette = "Paired", 
                    labels = c("P. septica", "P. agglomerans", "E. tasmaniensis", "E. amylovora", "T. ptyseos", "T. saanichensis", "E. cloacae",
                               "P. syringae")) +
  scale_y_continuous(limits = c(0, 9), breaks = c(0, 2, 4, 6, 8)) +
  labs(fill = "Closest Relative") +
  theme_bw() +
  theme(legend.text = element_text(face = "italic"),
        axis.title = element_blank(),
        panel.border = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        text = element_text(size = 12,  family = "Times New Roman"))

ggarrange(cal_NT, cal_AA, 
          labels = c("A", "B"),
          font.label = list(family = "Times New Roman"),
          label.x = 0.04,
          label.y = 0.98,
          common.legend = TRUE, 
          legend = "bottom")
```

In the AAS analysis, the genes closely related to the *Tatumella* species were not exclusive to the right half (Figure 3B). Additionally, the genes closely related to the *Erwinia* species and *E. cloacae* were more evenly distributed across the genome (Figure 3B). Similar to the NTS analysis, no gene clusters related to a single taxon were evident (Figure 3B).

## Closest Relative from Average Site Identity
Site identity is the amount of similarity between two sequences and accounts for the sequence length. The non-*Mixta* sequences were aligned to each of the *Mixta* homologs, so the percent identity was relative to *Mixta*. The closest relative to both *Mixta* species is defined as the taxon with the highest similarity to *Mixta*.

Similar to the closest relative according to genetic distances, *P. septica* was the closest relative for most *Mixta* sequences (Figure 4). With the NTS, *P. septica* was the closest relative for 79.3 and 80.6% of the *M. calida* and *M. gaviniae* sequences, respectively (Figure 4). However, these percentage decreased by 31.5 and 33.0%, respectively, with the AAS (Figure 4). The *Mixta* AAS were instead more closely related to other taxa except for *P. syringae* (Figure 4). Unexpectedly, 1.9 and 1.3% of the *M. calida* and *M. gaviniae* sequences, respectively, were most similar to the *E. cloacae* sequences (Figure 4). Additionally, these percentages increased by 3.1 and 2.5% for *M. calida* and *M. gaviniae*, respectively, for the AAS (Figure 4).

```{r Figure4, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="Figure 4. The percentage of genes (n = 799) that were most closely related to any of the eight non-*Mixta* species using percent site identity. The species with the highest percent identity to the *Mixta* species is considered to be the closest relative. The percentage of *Mixta* genes that were most closely related to any of the non-*Mixta* species are represented by the blue and green bars for the nucleotide and amino acid sequences, respectively. The lighter shades represent *M. calida* and the darker shades represent *M. gaviniae*.", fig.width=6.5}
number_relatives <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure4_FirstRelativeNTAA_Identity.csv", 
                             stringsAsFactors = FALSE) %>%
  mutate(Species = factor(Species, levels = c("P. septica", "P. agglomerans", "E. tasmaniensis", "E. amylovora", "T. ptyseos", "T. saanichensis", 
                                              "E. cloacae", "P. syringae"), ordered = TRUE),
         Type = factor(Type, levels = c("calida_NT", "gaviniae_NT", "calida_AA", "gaviniae_AA"), ordered = TRUE))


ggplot(data = number_relatives, aes(x = Species, y = Percent, fill = Type)) +
  geom_bar(stat = "identity", position = position_dodge()) +
  scale_fill_brewer(palette = "Paired", 
                    labels = c(expression(paste("NT and ", italic("M. calida"), sep = "")), 
                               expression(paste("NT and ", italic("M. gaviniae"), sep = "")), 
                               expression(paste("AA and ", italic("M. calida"), sep = "")), 
                               expression(paste("AA and ", italic("M. gaviniae"), sep = "")))) +
  labs(y = "Percent of Genes", 
       fill = expression(paste("Sequence type and ", italic("Mixta"), " species", sep = ""))) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, face = "italic"),               # Rotates and italicizes x axis labels
        legend.position = c(0.95, 0.95),                                                  # Moves legend inside the plot and in the top-right corner
        legend.justification = c("right", "top"),                                         # Top right corner is at above coordinates
        legend.text.align = 0,                                                            # Aligns legend text to the left
        legend.margin = margin(6, 6, 6, 6),                                               # Margins around legend
        text = element_text(size = 12,  family = "Times New Roman"))

```

When comparing the site identity analysis results, 89.2% of the NTS and 85.2% of the AAS agreed on the same closest relative between the two *Mixta* species. In contrast, only 55.1 and 53.9% of the *M. calida* and *M. gaviniae* sequences, respectively, had the same results between the NTS and AAS analyses.

## Spatial Patterns from Average Site Identity
Positional effects were investigated in the site identity results. Only the results of *M. calida* are shown because the *M. gaviniae* results showed similar patterns. The genes most similar to either of the *Pantoea* species and *E. cloacae* were evenly dispersed across the *Mixta* chromosome for both the NTS and AAS (Figure 5A and 5B). Similar to the genetic distance results, genes that are labelled as being closely related to either of the *Tatumella* species are located only on right side of chromosome, and genes that are closely related to *E. amylovora* are predominantly on the right side as well (Figure 5A). Unexpectedly, the lower left quadrant of the chromosome contains primarily genes that are closely related to either of the *Pantoea* species with only a few exceptions (Figure 5A). 	In the AAS analysis, genes closely related to any of the other taxa were more evenly distributed across the genome (Figure 5B).

```{r Figure5, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="Figure 5. The closest relatives to *M. calida* genes for (**A**) nucleotide and (**B**) amino acid sequences around the *M. calida* chromosome. The length and colour of the bars represent the different species to which the gene is most closely related according to site identity. The *M. calida* genome has 4092 genes.", fig.width=6.5, fig.height=3.9}
FR_nucl_calida <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure5A_Circular_NTcalida_Identity.csv", 
                           stringsAsFactors = FALSE) %>%
  mutate(Closest_Relative = factor(Closest_Relative, 
                                   levels = c("P_septica", "P_agglomerans", "E_tasmaniensis", "E_amylovora", "T_ptyseos", 
                                              "T_saanichensis", "E_cloacae", "P_syringae"), ordered = TRUE))

FR_aa_calida <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure5B_Circular_AAcalida_Identity.csv", 
                         stringsAsFactors = FALSE) %>%
  mutate(Closest_Relative = factor(Closest_Relative, 
                                   levels = c("P_septica", "P_agglomerans", "E_tasmaniensis", "E_amylovora", "T_ptyseos", 
                                              "T_saanichensis", "E_cloacae", "P_syringae"), ordered = TRUE))

cal_NT <- ggplot(FR_nucl_calida, aes(x = ID, y = Relative_Number, fill = Closest_Relative)) +
  geom_bar(stat = "identity", position = position_dodge(), width = 20) +
  coord_polar() +
  scale_fill_brewer(palette = "Paired", 
                    labels = c("P. septica", "P. agglomerans", "E. tasmaniensis", "E. amylovora", "T. ptyseos", "T. saanichensis", "E. cloacae",
                               "P. syringae")) +
  scale_y_continuous(limits = c(0, 9), breaks = c(0, 2, 4, 6, 8)) +
  labs(fill = "Closest Relative") +
  theme_bw() +
  theme(legend.text = element_text(face = "italic"),
        axis.title = element_blank(),
        panel.border = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        text = element_text(size = 12,  family = "Times New Roman"))

cal_AA <- ggplot(FR_aa_calida, aes(x = ID, y = Relative_Number, fill = Closest_Relative)) +
  geom_bar(stat = "identity", position = position_dodge(), width = 20) +
  coord_polar() +
  scale_fill_brewer(palette = "Paired", 
                    labels = c("P. septica", "P. agglomerans", "E. tasmaniensis", "E. amylovora", "T. ptyseos", "T. saanichensis", "E. cloacae",
                               "P. syringae")) +
  scale_y_continuous(limits = c(0, 9), breaks = c(0, 2, 4, 6, 8)) +
  labs(fill = "Closest Relative") +
  theme_bw() +
  theme(legend.text = element_text(face = "italic"),
        axis.title = element_blank(),
        panel.border = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        text = element_text(size = 12,  family = "Times New Roman"))

ggarrange(cal_NT, cal_AA, 
          labels = c("A", "B"),
          font.label = list(family = "Times New Roman"),
          label.x = 0.04,
          label.y = 0.98,
          common.legend = TRUE, 
          legend = "bottom")
```

## A Comparison of the Four Analyses
The four above analyses (genetic distance and site identity, both using NTS and AAS) gave slightly different results for the *Mixta* sequences. When comparing the results, approximately 38.7% of both the *M. calida* and *M. gaviniae* sequences agreed upon a closest relative. However, the number of sequences closely related to any of the other taxa were slightly different between *M. calida* and *M. gaviniae* (Table 3). There were no genes for which all four analyses agreed on *T. ptyseos* being the closest relative (Table 3). Additionally, 7.5% of the *M. calida* and *M. gaviniae* homologs did not share the same closest relative.

```{r Table3, echo=FALSE, message=FALSE, warning=FALSE, fig.cap=TRUE}
CR_calida <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Table3_AllResults_calida.csv", 
                      stringsAsFactors = FALSE)

# table(CR_calida$ALL)
# FALSE  TRUE         # FALSE: 61.3%
# 490    309          # TRUE: 38.7% of M. calida genes had the same closest relative in all four analyses

CR_gaviniae <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Table3_AllResults_gaviniae.csv", 
                        stringsAsFactors = FALSE)

# table(CR_gaviniae$ALL)
# FALSE  TRUE 
# 490    309

all_same_cal <- subset(CR_calida, ALL == TRUE); all_same_gav <- subset(CR_gaviniae, ALL == TRUE)

all_same_species <- data.frame(table(all_same_cal$NT_dist),
                               table(all_same_gav$NT_dist)) %>%
  subset(select = -Var1.1) %>%
  mutate(Var1 = factor(Var1, 
                          levels = Spp, ordered = TRUE)) %>%
  arrange(Var1)

colnames(all_same_species) <- c("Species", "# of M. calida genes", "# of M. gaviniae genes")

all_same_species$Species <- gsub("_", ". ", all_same_species$Species)

kable(all_same_species, 
      caption = "Table 3. The number of *M. calida* and *M. gaviniae* genes wherein all analyses gave the same species as the closest relative.")
```



```{r Table4, echo=FALSE, message=FALSE, warning=FALSE, fig.cap=TRUE}
CR_calida <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Table3_AllResults_calida.csv", 
                      stringsAsFactors = FALSE)

CR_gaviniae <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Table3_AllResults_gaviniae.csv", 
                        stringsAsFactors = FALSE)


cal_genus <- subset(CR_calida, Genus != "Nope")
gav_genus <- subset(CR_gaviniae, Genus != "Nope")

all_same_genus <- data.frame(table(cal_genus$Genus),
                             table(gav_genus$Genus))

all_same_genus <- subset(all_same_genus, select = -Var1.1)
colnames(all_same_genus) <- c("Genus", "# of M. calida genes", "# of M. gaviniae genes")

all_same_genus <- all_same_genus %>%
  mutate(Genus = factor(Genus, 
                       levels = c("Pantoea", "Erwinia", "Tatumella", "Enterobacter"), ordered = TRUE)) %>%
  arrange(Genus)

test <- data.frame(Gene = CR_calida$Gene, cal = CR_calida$Genus, gav = CR_gaviniae$Genus)
test$Same <- case_when(test$cal == test$gav ~ test$cal,
                       TRUE ~ "Nah")
test <- subset(test, Same == "Nah")
# table(test$Same)
# FALSE  TRUE 
# 39     760 

kable(all_same_genus, 
      caption = "Table 4. The number of *M. calida* and *M. gaviniae* genes wherein all analyses gave the same genus as the closest relative.")
```



```{r Figure6, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="Figure 5. The closest relatives to *M. calida* (**A**) and *M. gaviniae (**B**) genes that were agreed upon by all four analyses that were conducted in this study. The circular plot simulates the circular chromosome of the respective species. The colour of the bars represent the different species to which the gene is most closely related. The *M. calida* and *M. gaviniae* genomes have 4092 and 4242 genes, respectively.", fig.width=6.5, fig.height=3.9}
all_cal <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure6A_AllAgreement_calida.csv", stringsAsFactors = FALSE) %>%
  mutate(Agreement = factor(Agreement, levels = Spp, ordered = TRUE))

all_gav <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure6B_AllAgreement_gaviniae.csv", stringsAsFactors = FALSE) %>%
  mutate(Agreement = factor(Agreement, levels = Spp, ordered = TRUE))

calida <- ggplot(all_cal, aes(x = ID, y = 1, fill = Agreement)) +
  geom_bar(stat = "identity", position = position_dodge(), width = 5) + # 
  coord_polar() +
  scale_fill_brewer(palette = "Paired", 
                    labels = c("P. septica", "P. agglomerans", "E. tasmaniensis", "E. amylovora", "T. ptyseos", "T. saanichensis", "E. cloacae",
                               "P. syringae")) +
  scale_y_continuous(limits = c(0, 1), breaks = c(0, 1)) +
  labs(fill = "Closest Relative") +
  theme_bw() +
  theme(legend.text = element_text(face = "italic"),
        axis.title = element_blank(),
        panel.border = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        text = element_text(size = 12,  family = "Times New Roman"))

gaviniae <- ggplot(all_gav, aes(x = ID, y = 1, fill = Agreement)) +
  geom_bar(stat = "identity", position = position_dodge(), width = 5) + # 
  coord_polar() +
  scale_fill_brewer(palette = "Paired", 
                    labels = c("P. septica", "P. agglomerans", "E. tasmaniensis", "E. amylovora", "T. ptyseos", "T. saanichensis", "E. cloacae",
                               "P. syringae")) +
  scale_y_continuous(limits = c(0, 1), breaks = c(0, 1)) +
  labs(fill = "Closest Relative") +
  theme_bw() +
  theme(legend.text = element_text(face = "italic"),
        axis.title = element_blank(),
        panel.border = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        text = element_text(size = 12,  family = "Times New Roman"))

ggarrange(calida, gaviniae, 
          labels = c("A", "B"),
          font.label = list(family = "Times New Roman"),
          label.x = 0.04,
          label.y = 0.98,
          common.legend = TRUE, 
          legend = "bottom")
```


# Discussion

# Acknowledgements

# Contributions to the Field Statement

# Figure Legends

# Tables


\newpage
# References
<div id="refs"></div>
