---
title: "Untitled"
author: "Kimberly Hinz"
date: "2020/12/08"
output:
  word_document:
    reference_docx: Styles1.docx
    fig_caption: true
    # toc: true
    # toc_depth: 5
bibliography: References.bib
csl: IJSEM.csl
---

\newpage
# Abstract
*Mixta* is a recently described genus within the bacterial family *Erwiniaceae* that is closely related to *Pantoea*, *Erwinia*, and *Tatumella*. Depending on the genes and method used for phylogenetic analysis, the genus with which *Mixta* shares the most recent ancestor differs. This study aimed to determine the cause behind these contentious results, and identified genes that may have resulted from horizontal gene transfer. Ten complete or partial genomes – two *Mixta* species, six species from other *Erwiniaceae* genera, and two outgroups – were retrieved from NCBI and annotated. Homologous genes were extracted yielding a dataset of 799 genes. The genes were aligned With the ClustalW algorithm and MEGA-CC was used for model testing and computing phylogenetic trees for each gene for both nucleotide and amino acid sequences. Distance matrices were then extracted from these trees. Nucleotide and amino acid identity analyses were also done in `R` programming language. *Pantoea* was the closest relative to the *Mixta* species in most of the analyses; however, results were not consistent across analyses. Some genes were also found to be more similar to other, non-*Pantoea* species. Diligence must be given to the selection of genes for phylogenetic analysis and to the method chosen to prevent xenologous signal from distorting the true relationships. Furthermore, future research should consider that different phylogenetic analyses may give different results.

\newpage
# Introduction
The bacterial family Erwiniaceae includes diverse Gram-negative [@Winslow1917; @Prakash2015] organisms including the genera *Mixta*, *Pantoea*, *Erwinia*, *Tatumella*, *Buchnera*, *Wigglesworthia*, *Phaseolibacter*, *Kalamiella* (LPSN, http://www.bacterio.net/). Three of these, *Mixta*, *Pantoea*, and *Erwinia*, have been the focus of numerous studies given their relevance to plant and human pathogens. For example, the genus *Erwinia* contains plant pathogens (e.g. E. amylovora) [@HaubenSwings2005; @Lopez2011; @Matsuura2012], epiphytes (e.g. E. tasmaniensis) [@Geider2006; @Mergaert1999], species that may contribute to plant disease [@Buonaurio2015], and others that are insect-associated [@Campillo2015; @Harada1997; @Skrodenyte-Arbaciauskiene2012]. Similarly, *Pantoea*, a genus reported by Gavini et al.[@Gavini1989], includes species that inhabit a variety of environments [@Gavini1989; @Popp2010; @Liu2013], including human and plant pathogens [@Prakash2015; @deBaere2004; @Fullerton2007; @Kratz2003; @Lim2006; @Schmid2003; @vanRostenberghe2006] as well as species associated with plant growth [@Smits2011; @Kim2012], insects [@Palmer2016], and fungi [@Ma2016; @Rong2016]. *Mixta*, originally *Pantoea*, is a novel genus first suggested by Palmer et al. [@Palmer2017] and then later proposed by the same researchers in 2018 [@Palmer2018]. Two species, M. calida and M. gaviniae (previously species of *Pantoea*), were originally found in infant formula and its production environment [@Popp2010]. Fritz et al. [@Fritz2014] demonstrated that some strains of M. calida were linked with post-surgical meningitis in humans. Currently, there are only five *Mixta* species validly published (LPSN, http://www.bacterio.net/).

Understanding the phylogeny of bacterial species and genes allows researchers to investigate the evolutionary history of the species, their relatedness, and the individual genes that allow the species to survive under various conditions [@Palmer2017; @Bennett2012; @Prasanna2013; @Angus2014; @deMaayer2014; @Fouts2016]. The most common methods in phylogenetic analysis are those including 16S ribosomal RNA (rRNA) sequences [@Konstantinidis2007], multi-locus sequence analysis (MLSA) with several housekeeping genes [@Konstantinidis2007; @Gevers2005; @Glaeser2015], and ribosomal MLSA with 53 structural ribosomal proteins [@Palmer2017; @Bennett2012; @Jolley2012]. However, the phylogenetic trees generated from these methods may not be very robust [@Konstantinidis2007; @Glaeser2015; @Brady2008] because of the lower phylogenetic signal from the highly conserved housekeeping genes [@Konstantinidis2007; @Gevers2005; @Fox1992; @Konstantinidis2005; @Staley2006; @RichterRosselloMora2009]. This is particularly seen within Erwiniaceae wherein most of our understanding of the phylogeny and interrelationships are based on the 16S rRNA genes [@Francino2006; @Hauben1998; @Naum2008; @Sproer1999; @Adeolu2016] because of its ubiquity. However, it has insufficient resolution at the species level [@Glaeser2015] and the genera *Pantoea*, *Erwinia*, *Tatumella*, and *Mixta* are often not monophyletic [@Rezzonico2016]. Another issue that may arise with MLSA is unknowingly including paralogues (i.e. genes originating from an intragenomic duplication event) or xenologues (i.e. genes originating from horizontal gene transfer) [@Palmer2017; @Koonin2005].

The DNA topoisomerase (gyrase) subunit (gyrB), DNA-directed RNA polymerase subunit (rpoB), ATP synthase subunit (atpD), and initiation translation factor (infB) have been shown to be more reliable than 16S rRNA gene sequences for determining intra- and intergeneric relationships [@Hedegaard1999; @Dauga2002; @Mollet1997; @Paradis2005] and a robust method for the Erwiniaceae family [@Brady2010]. The genera within Erwiniaceae are monophyletic when concatenated partial gyrB, rpoB, atpD, and infB gene sequences are used [@Brady2008; @Brady2009; @Moretti2011].

Nevertheless, despite efforts to find a suitable method for identifying the species tree for Erwiniaceae genera, *Mixta*’s position with relation to *Pantoea*, *Erwinia*, and *Tatumella* vastly differs between studies [@Prakash2015; @Palmer2017; @Palmer2018; @Adeolu2016; @Brady2011; @Rezzonico2016; @Brady2012; @Gueule2015]. Not only do different gene combinations provide different results, but different algorithms and models for alignments and phylogenetic trees also contribute to the contentious results [@Naum2008; @Octavia2014]. For example, Prakash et al. [@Prakash2015], Palmer et al. [@Palmer2017], Rezzonico et al. [@Rezzonico2016], and Gueule et al. [@Gueule2015] all used gyrB, rpoB, atpD, and infB nucleotide gene sequences in their phylogenetic analyses. Prakash et al. conducted a neighbour-joining statistical method while the others used a maximum-likelihood (ML) method. Three of the studies – Rezzonico et al., Prakash et al., and Gueule et al. – placed the *Mixta* species within the *Pantoea* genera whereas the analysis in Palmer et al. placed *Mixta* outside of the other three genera. Palmer et al. [@Palmer2018], Rezzonico et al. [@Rezzonico2016], and Brady et al. [@Brady2012] also ran analyses with the four MLSA genes as amino acid sequences. These three studies show contentious results as Rezzonico et al. found that *Mixta* is related closest with *Tatumella*, Palmer et al. found that *Mixta* is related closest with *Pantoea*, and Brady et al. found *Mixta* within the *Pantoea* genus. In a separate study conducted by Palmer et al. @Palmer2017, two ML trees were computed, one with 1039 core amino acid sequence and the other with 52 ribosomal MLSA nucleotide genes. Both ML trees contained species from *Mixta*, *Pantoea*, *Erwinia*, and *Tatumella*; however, the analysis with the core genes placed *Pantoea* and *Tatumella* as sister taxa followed by two *Mixta* species whereas the analysis with ribosomal MLSA genes placed *Pantoea* and *Tatumella* as sister taxa followed by *Erwinia* and then by the *Mixta* species. 

There are two possible causes behind contentious results in phylogenetic analyses. One is the use of different methods, including the use of a different phylogenetic tree model from another study despite both studies conducting maximum likelihood trees and the same sequence type. Additionally, there is no common phylogenetic analysis model between nucleotide and amino acid sequences; therefore, evolutionary history is estimated differently. The other reason may be horizontal gene transfer (HGT) events within the *Erwiniaceae* or from a non-closely-related species. HGT events occur frequently within bacteria and often the genes provide the bacteria the ability to adapt to new environments and/or survive selective pressures.

The objective of this paper was to determine the cause behind the contentious phylogenetic results for the *Mixta* species. To accomplish this, homologous genes were extracted from ten complete or partial genomes - two for each of *Mixta*, *Pantoea*, *Erwinia*, and *Tatumella*, and two outgroups - that were retrieved from public databases. Model testing was done for each gene in the dataset and phylogenetic trees were computed using the recommended model. Distance matrices were exported from the phylogenetic trees and these were used to determine which species were most closely related to *Mixta* for each gene. Nucleotide and amino acid identity analyses were also conducted for each gene in order to compare 

The objective of my honours thesis was to determine the cause behind the contentious phylogenetic results for *Mixta* species. To accomplish this, ten complete or partial genomes – two each for *Mixta*, *Pantoea*, *Erwinia*, *Tatumella*, and outgroup – were retrieved from public databases, annotated, and homologous genes were extracted. Genes were used to make individual phylogenetic trees, and the trees were examined for inconsistent evolutionary histories.

# Materials and Methods
## Genomes and Gene Sets
Publicly available full genomes were retrieved from NCBI comprising of ten type strains from *Mixta*, Pantoae, *Erwinia*, *Tatumella*, Citrobacter, and Enterobacter (Table 1). The genomes were annotated with Prokka v1.14.1 61 using default parameters and homologous genes were extracted using GET_HOMOLOGUES software package 62 with the bidirectional best-hit search algorithm using default parameters. GET_HOMOLOGUES extracted 1284 genes that were common between the ten species.
Within R 63, a filtering function was created to pass forward fasta files with ten, non-truncated sequences. The minimum gene length to pass through the filter was 90% of the length of the longest sequence in that file. This filter ensured that files containing split, missing, or paralogous genes were not included in the analysis. As a result, 1035 files of the 1284 passed this filtering stage.

## Gene Alignment
Gene alignments were done in R with a few software packages. The fasta files were read in using the readDNAStringSet() function 64 (package: Biostrings). The alignments were made using the ClustalW algorithm with the msaClustalW() 65 (package: msa). The number of maximum iterations was set to 100 and the other parameters were kept at default settings 65. Alignments were then converted into a readable format using msaConvert 65 (package: msa) and then written as fasta files using write.fasta() 66 (package: seqinr).

## Distance Matrices
Prior to computing distance matrices, model testing was done for each of the 1035 gene files through MEGA-CC 67 by calling a text file containing the file pathways and a MAO file detailing the analysis preferences. These results were exported as comma-separated values (CSV) and read into R. The best model was chosen based on the lowest Bayesian Information Criterion (BIC) and if it was available for distance matrices in MEGAX 68. General Time Reversible (GTR) models and Hasegawa-Kishino-Yano (HKY) models were not available for distance matrices and so these models were not considered during model testing. Furthermore, the rates among sites option invariant sites (+I) was also not available for distance matrices and were subsequently ignored.

The gene files were separated into groups according to the suggested model and text files were created with their file pathways. These text files were called in turn along with a corresponding MAO file through MEGA-CC to compute the distance matrices for each gene. Variance was estimated with 500 bootstraps, transitions and transversions were allowed, and gaps and missing data were treated with pairwise deletion.

The genetic distances and standard errors relative to M. calida and M. gaviniae were extracted to create plots in R, categorize the closest relatives, run generalized additive models to determine potential non-linear relationships between parameters and genetic distances, and identify genes for further analysis using phylogenetic trees. The ML phylogenetic trees were created using MEGAX and with a model suggested through model testing.

# Results and Discussion

# Conclusion

```{=openxml}
<w:p><w:r><w:br w:type="page"/></w:r></w:p>
```
# Literature Cited
<div id="refs"></div>

\newpage
# Appendix