---
title: "An investigation into the contentious phylogenetic results of the *Mixta* species"
output:
  word_document:
    reference_docx: Style_Manuscript.docx
    fig_caption: true
    # toc: true
    # toc_depth: 5
bibliography: References.bib
csl: IJSEM.csl
---
```{r Packages, echo = FALSE, message = FALSE, warning=FALSE}
library("knitr")
library("tidyverse")
library("ggplot2")
library("RColorBrewer")
library("extrafont")
theme_set(theme_classic())
```

Kimberly N. Hinz, John Stavrinides^1^

Department of Biology  
University of Regina  
3737 Wascana Parkway  
Regina, Saskatchewan, Canada  
S4S0A2

^1^**Correspondence**:  
John Stavrinides  
Department of Biology  
University of Regina  
3737 Wascana Parkway  
Regina, Saskatchewan  
S4S0A2  
Ph: (306) 337-8478  
Email: john.stavrinides@uregina.ca

**Keywords:**

<!-- \newpage -->
# Abstract
<!-- *Mixta* is a recently described genus within the bacterial family *Erwiniaceae* and is closely related to *Pantoea*, *Erwinia*, and *Tatumella*. Depending on the genes and method used for phylogenetic analysis, the genus with which *Mixta* shares the most recent ancestor differs. This study aimed to determine the cause behind these contentious results and identified genes that may have resulted from horizontal gene transfer. Ten complete or partial genomes – two *Mixta* species, six species from other *Erwiniaceae* genera, and two outgroups – were retrieved from NCBI and annotated. Homologous genes were extracted yielding a dataset of 799 genes. The genes were aligned with the ClustalW algorithm, and MEGA-CC was used for model testing and computing phylogenetic trees for each gene for both nucleotide and amino acid sequences. Distance matrices were then extracted from these trees. Nucleotide and amino acid identity analyses were also done using the programming language `R`. *Pantoea* was the closest relative to the *Mixta* species in most analyses; however, results were not consistent. Some genes were also found to be more similar to other, non-*Pantoea* species. Diligence must be given to selecting genes for phylogenetic analysis and the method chosen to prevent any xenologous signal from distorting the actual relationships. Furthermore, future research should consider that different phylogenetic analyses may provide different results. -->

# Intro and MM
<!-- \newpage -->
<!-- # Introduction -->
<!-- The bacterial family *Erwiniaceae* includes diverse Gram-negative [@Winslow1917; @Prakash2015] organisms, including the genera *Mixta*, *Pantoea*, *Erwinia*, *Tatumella*, *Buchnera*, *Wigglesworthia*, *Phaseolibacter*, and *Kalamiella* [REF CRAIG]. *Mixta* is a relatively newer genus first identified by Palmer et al. in 2017 [@Palmer2017] and later described by the same researchers in 2018 [@Palmer2018]. Currently, there are only five *Mixta* species validly published [@LPSN]. *M. calida* and *M. gaviniae* (previously species of *Pantoea*) were initially found in infant formula and its production environment [@Popp2010]. Fritz et al. [@Fritz2014] demonstrated that some strains of *M. calida* were linked with post-surgical meningitis in humans. The other *Mixta* species include *M. theicola* which was isolated from black tea extract [@Kato2015], *M. intestinalis* which was isolated from a healthy human fecal sample [@Prakash2015], and *M. tenebrionis* which was isolated from the gut microbiota of a plastic-eating mealworm [@Xia2020]. The first two were previously named as *Pantoea* species. -->

<!-- *Mixta*'s position relative to other members of the *Erwiniaceae*, in particular to *Pantoea*, *Erwinia*, *Tatumella*, differs vastly between studies [@Prakash2015; @Palmer2017; @Palmer2018; @Adeolu2016; @Brady2011; @Rezzonico2016; @Brady2012; @Gueule2015]. Prakash et al. [@Prakash2015], Palmer et al. [@Palmer2017], Rezzonico et al. [@Rezzonico2016], and Gueule et al. [@Gueule2015] all used *gyrB*, *rpoB*, *atpD*, and *infB* nucleotide gene sequences in their phylogenetic analyses. Prakash et al. [@Prakash2015] conducted a neighbour-joining statistical method while the others used a maximum-likelihood (ML) method [@Palmer2017; @Rezzonico2016; @Gueule2015]. Three of the studies – Rezzonico et al. [@Rezzonico2016], Prakash et al. [@Prakash2015], and Gueule et al. [@Gueule2015] – placed the *Mixta* species within the *Pantoea* genera. In contrast, the analysis in Palmer et al. [@Palmer2017] identified *Mixta* outside of the other three genera. A separate study by Palmer et al. [@Palmer2018], Rezzonico et al. [@Rezzonico2016], and Brady et al. [@Brady2012] also ran analyses with the four MLSA genes as amino acid sequences. *Mixta* was found to be related closest to *Tatumella* by Rezzonico et al. [@Rezzonico2016] and *Pantoea* by Palmer et al. [@Palmer2018]. Brady et al. [@Brady2012] suggested that *Mixta* was a part of the *Pantoea* genus and not separate. Additionally, Palmer et al. [@Palmer2017] computed two ML trees; one used 1039 core amino acid sequences, and the other used 52 ribosomal MLSA nucleotide sequences. The 1039 core gene tree suggested *Pantoea* and *Tatumella* were sister taxa followed by two *Mixta* species. However, the ribosomal MLSA gene tree proposed *Pantoea* and *Tatumella* were sister taxa followed by *Erwinia* and then by *Mixta*. -->

<!-- What remains unclear is the relative contribution of disaparate evolutionary histories of *Mixta* genes, tree-building methods, and sequence evolution on the phylogenetic position of the genus. This paper aims to determine the cause behind the *Mixta* species' contentious phylogenetic results. Ten complete or partial genomes – two for each of *Mixta*, *Pantoea*, *Erwinia*, and *Tatumella* and two outgroups – were retrieved from a public database. Homologous genes were extracted from these genomes in both the nucleotide and amino acid sequences. Distance matrices were exported from the phylogenetic trees, and these were used to determine which species were most closely related to *Mixta* for each gene. Nucleotide and amino acid identity analyses were also conducted for each gene to compare the results between evolutionary history and sequence similarity. Phylogenetic trees were constructed for combinations of different gene sequences.  -->

<!-- # Materials and Methods -->
<!-- ## Genomes and Gene Sets -->
<!-- Ten publicly available full or partial genomes of the species' type strains were retrieved from the National Center for Biotechnology Information (NCBI). Two representative genomes were used for each of the four *Erwiniaceae* genera while *Enterobacter cloacae* (*Enterobacterales*) and *Pseudomonas syringae* (*Pseudomonaceae*) were used as outgroups (Table 1). The genomes were annotated with Prokka v1.14.1 [@Seemann2014] using default parameters, and homologous genes were extracted using GET_HOMOLOGUES software package [@Contreras-Moreira2013] with the bidirectional best-hit search algorithm using default parameters. GET_HOMOLOGUES extracted 954 homologous genes between the ten species. -->

<!-- ```{r Table1, echo=FALSE, message=FALSE, warning=FALSE, fig.cap=TRUE} -->
<!-- strain <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Table1_Genomes.csv", stringsAsFactors = FALSE) -->

<!-- colnames(strain)[3] <- "GenBank Accession No." -->

<!-- kable(strain, caption = "Table 1. Genomes used in this study.") -->
<!-- ``` -->

<!-- In `R` [@RCore2019], two filter functions were written for the nucleotide sequence gene files. The first filter kept genes files that contained ten, single-copy sequences that were approximately the same length. The shortest gene length allowed was 90% the length of the longest sequence in the gene file. Of the 954 homologous gene sets, 38 multi-copy genes were excluded, and another 175 were excluded for not meeting the length requirements.  -->

<!-- A second filter was applied to these 175 genes to identify any genes wherein both representatives of each genera were approximately the same length. The filter split the genes into two groups: sequences that met the original 90% length cutoff and sequences that were between 80 and 90% the length of the longest sequence. If both representatives of a genus were in the same group, then the gene was added to the working dataset. If at least one of the sequences did not meet the 80% cutoff, then the gene was excluded. An additional 58 genes were added to the working dataset, resulting in 799 genes. The amino acid sequences for these genes were subsequently downloaded. -->

<!-- ## Gene Alignment -->
<!-- Gene alignments were done in `R` using a ClustalW algorithm. The fasta files were read into the session using the `readDNAStringSet()` function (package: Biostrings) [@Pages2020]. The function called to do the alignments was the `msaClustalW()` function (package: msa) [@Bodenhofer2015] using default parameters and 100 iterations. The aligned sequences were then converted into a comma-separated values (CSV)-compatible format using `msaConvert()` (package: msa)[@Bodenhofer2015] and written into a new fasta file using the `write.fasta()` function (package: seqinr) [@Charif2007] -->

<!-- ## Model Selection -->
<!-- The best model of sequence evolution was identified for each nucleotide and amino acid alignment using Model Selection as implemented in MEGA-CC [@Kumar2012]. Default parameters were used for model testing. The output was a CSV file for each gene. These CSV files were read into `R`, and the model for phylogenetic analysis was extracted based on the lowest Bayesian Information Criterion (BIC). -->

<!-- ## Phylogenetic Analysis -->
<!-- Phylogenetic analysis was performed in MEGA-CC using maximum likelihood with 500 bootstrap replications, the model, and rate patterns to create a phylogenetic tree for each gene. If a model required a Gamma distribution, the number of discrete Gamma categories was set to 5, which is the default. -->

<!-- Distance matrices were extracted from each of the phylogenetic trees and saved as Excel files. These files were then read into the `R` session. The genetic distances between *Mixta* and the other species were extracted for further analysis. Tables were created in `R Markdown` using the `kable()` function (package: knitr) [@Xie2020] and figures were created using the `ggplot()` function (package: ggplot2) [@Wickham2016]. -->

<!-- ## Nucleotide and Amino Acid Identity -->
<!-- Each of the non-*Mixta* sequences were aligned to one of the corresponding *Mixta* sequences in turn. Alignments were performed in R as described above, and the aligned sequences were then split into the individual sites using the `str_split()` function (package: stringr) [@Wickham2019]. A function was created using base `R` code to calculate the number of identical sites between each sequence and the *Mixta* homologs. Values that equal or are close to 100% are more identical to the *Mixta* sequences. -->

# Results
```{r Table2, echo=FALSE, message=FALSE, warning=FALSE, fig.cap=TRUE}
models <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Table2_BestAvailableModels.csv", stringsAsFactors = FALSE)

colnames(models)[2:3] <- c("Best Model", "Best Available Model")

kable(models, 
      caption = "Table 2. The number of genes that required each model according to the best model with the lowest BIC and the best available model for genetic distance analysis.")
```

```{r Figure1, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="Figure 1. The percentage of genes (n = 799) that were most closely related to any of the eight non-*Mixta* species when using the recommended model from model testing versus the best available) model for genetic distance analysis in MEGAX. The sequences were nucleotide sequences. The species with the shortest genetic distance from the *Mixta* species is considered to be the closest relative. The pink and orange bars represent the percentage of *Mixta* genes that are most closely related to any of the non-*Mixta* species when using the recommended model and the best available model, respectively. The lighter shades represent *M. calida* and the darker shades represent *M. gavinaie*.", fig.width=6.5}
number_relatives <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure1_FirstRelativeBestvsAvailableModelNT.csv", 
                             stringsAsFactors = FALSE) %>%
  mutate(Species = factor(Species, levels = c("P. septica", "P. agglomerans", "E. tasmaniensis", "E. amylovora", "T. ptyseos", "T. saanichensis", 
                                              "E. cloacae", "P. syringae"), ordered = TRUE),
         Type = factor(Type, levels = c("PD_calida", "PD_gaviniae", "GD_calida", "GD_gaviniae"), ordered = TRUE))

ggplot(data = number_relatives, aes(x = Species, y = Percent, fill = Type)) +
  geom_bar(stat = "identity", position = position_dodge()) +
  scale_fill_manual(values = c("#fb9a99", "#e31a1c", "#fdbf6f", "#ff7f00"),
                    labels = c(expression(paste("Recommended Model and ", italic("M. calida"), sep = "")), 
                               expression(paste("Recommended Model and ", italic("M. gaviniae"), sep = "")), 
                               expression(paste("Best Available Model and ", italic("M. calida"), sep = "")), 
                               expression(paste("Best Available Model and ", italic("M. gaviniae"), sep = "")))) +
  labs(y = "Percent of Genes", 
       fill = expression(paste("Model and ", italic("Mixta"), " species", sep = ""))) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, face = "italic"),               # Rotates and italicizes x axis labels
        legend.position = c(0.95, 0.95),                                                  # Moves legend inside the plot and in the top-right corner
        legend.justification = c("right", "top"),                                         # Top right corner is at above coordinates
        legend.text.align = 0,                                                            # Aligns legend text to the left
        legend.margin = margin(6, 6, 6, 6),                                               # Margins around legend
        text = element_text(size = 12,  family = "Times New Roman"))
```

























<!-- # Old Results -->
<!-- ```{r Table2, echo=FALSE, message=FALSE, warning=FALSE, fig.cap=TRUE} -->
<!-- PM_NT <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Table2_PmodelsNT.csv", stringsAsFactors = FALSE) -->

<!-- kable(PM_NT, caption = "Table 2. According to model testing and the lowest BIC for nucleotide sequences, the number of genes required each phylogenetic tree model.") -->
<!-- ``` -->

<!-- ```{r Table3, echo = FALSE, message = FALSE, warning=FALSE, fig.cap=TRUE} -->
<!-- PM_AA <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Table3_PmodelsAA.csv", stringsAsFactors = FALSE) -->

<!-- kable(PM_AA, caption = "Table 3. According to model testing and the lowest BIC for amino acid sequences, the number of genes required each phylogenetic tree model.") -->
<!-- ``` -->

<!-- ```{r Figure1, echo = FALSE, message = FALSE, warning=FALSE, fig.cap="Figure 1. The percent of genes (n = 799) requiring each model when using the recommended model for phylogenetic tree analysis (green) and when using the best available model for genetic distance analysis (orange). Models are for nucleotide sequences.", fig.width=6.5} -->
<!-- models <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure1_BestvAvailableModels.csv", stringsAsFactors = FALSE) -->

<!-- ggplot(data = models, aes(x = Model, y = Percent, fill = Bst_Avlb)) + -->
<!--   geom_bar(stat = "identity", position = position_dodge()) + -->
<!--   scale_fill_brewer(palette = "Dark2", -->
<!--                     labels = c("Phylogenetic", "Genetic Distance")) + -->
<!--   labs(y = "Percent of Genes \n Requiring Each Model", -->
<!--        fill = "Analysis") + -->
<!--   theme(axis.text.x = element_text(angle = 45, hjust = 1),                                # Rotates x axis labels -->
<!--         legend.position = c(0.3, 0.95),                                                   # Moves legend inside the plot and in the top-left corner -->
<!--         legend.justification = c("right", "top"),                                         # Top right corner is at above coordinates -->
<!--         legend.text.align = 0,                                                            # Aligns legend text to the left -->
<!--         legend.margin = margin(6, 6, 6, 6),                                               # Margins around legend -->
<!--         text = element_text(size = 12,  family = "Times New Roman")) -->
<!-- ``` -->

<!-- ```{r Figure2, echo = FALSE, message = FALSE, warning=FALSE, fig.cap="Figure 2. The percentage of genes (n = 799) most closely related to any of the eight non-*Mixta* species. Closest relative was inferred using shortest genetic distance. Blue bars represent the nucleotide sequences and green bars represent the amino acid sequences. The lighter of each colour represents *M. calida* and the darker *M. gaviniae*.", fig.width=6.5} -->
<!-- number_relatives <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure2_FirstRelativeNTAA.csv", -->
<!--                              stringsAsFactors = FALSE) %>% -->
<!--   mutate(Species = factor(Species, levels = c("P. septica", "P. agglomerans", "E. tasmaniensis", "E. amylovora", "T. ptyseos", "T. saanichensis", -->
<!--                                               "E. cloacae", "P. syringae"), ordered = TRUE), -->
<!--          Type = factor(Type, levels = c("calida_NT", "gaviniae_NT", "calida_AA", "gaviniae_AA"), ordered = TRUE)) -->


<!-- ggplot(data = number_relatives, aes(x = Species, y = Percent, fill = Type)) + -->
<!--   geom_bar(stat = "identity", position = position_dodge()) + -->
<!--   scale_fill_brewer(palette = "Paired", -->
<!--                     labels = c(expression(paste("NT and ", italic("M. calida"), sep = "")), -->
<!--                                expression(paste("NT and ", italic("M. gaviniae"), sep = "")), -->
<!--                                expression(paste("AA and ", italic("M. calida"), sep = "")), -->
<!--                                expression(paste("AA and ", italic("M. gaviniae"), sep = "")))) + -->
<!--   labs(y = "Percent of Genes", -->
<!--        fill = expression(paste("Sequence type and ", italic("Mixta"), " species", sep = ""))) + -->
<!--   theme(axis.text.x = element_text(angle = 45, hjust = 1, face = "italic"),               # Rotates and italicizes x axis labels -->
<!--         legend.position = c(0.95, 0.95),                                                  # Moves legend inside the plot and in the top-right corner -->
<!--         legend.justification = c("right", "top"),                                         # Top right corner is at above coordinates -->
<!--         legend.text.align = 0,                                                            # Aligns legend text to the left -->
<!--         legend.margin = margin(6, 6, 6, 6),                                               # Margins around legend -->
<!--         text = element_text(size = 12,  family = "Times New Roman")) -->
<!-- ``` -->

<!-- ```{r Figure2, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="Figure 2. The percentage of genes (out of 799) that were most closely related to any of the eight non-*Mixta* species when using the recommended (best) model from model testing vs the restricted model (best available) options available for genetic distance analysis in MEGAX for nucleotide sequences. The species with the shortest genetic distance from both of the *Mixta* species is considered the closest relative. Evolutionary patterns may vary by gene; therefore, it is appropriate to conduct model testing and use the model that explains the most variation. Nevertheless, this model may not be available in software programs or analysis methods within a software program. Pink bars represent the percentage of *Mixta* genes most closely related to non-*Mixta* species when using the recommended model. Orange bars represent the percentage of *Mixta* genes most closely related to non-*Mixta* species when using the best available genetic distance model analysis in MEGAX. GTR and HKY models and invariant sites parameter were not available for this analysis. The lighter shades represent *M. calida*, and the darker shades represent *M. gaviniae*.", fig.width=6.5} -->
<!-- number_relatives <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure2_FirstRelativeBestvsAvailableModelNT.csv", -->
<!--                              stringsAsFactors = FALSE) %>% -->
<!--   mutate(Species = factor(Species, levels = c("P. septica", "P. agglomerans", "E. tasmaniensis", "E. amylovora", "T. ptyseos", "T. saanichensis", -->
<!--                                               "E. cloacae", "P. syringae"), ordered = TRUE), -->
<!--          Type = factor(Type, levels = c("PD_calida", "PD_gaviniae", "GD_calida", "GD_gaviniae"), ordered = TRUE)) -->

<!-- ggplot(data = number_relatives, aes(x = Species, y = Percent, fill = Type)) + -->
<!--   geom_bar(stat = "identity", position = position_dodge()) + -->
<!--   scale_fill_manual(values = c("#fb9a99", "#e31a1c", "#fdbf6f", "#ff7f00"), -->
<!--                     labels = c(expression(paste("Best Model and ", italic("M. calida"), sep = "")), -->
<!--                                expression(paste("Best Model and ", italic("M. gaviniae"), sep = "")), -->
<!--                                expression(paste("Best Available Model and ", italic("M. calida"), sep = "")), -->
<!--                                expression(paste("Best Available Model and ", italic("M. gaviniae"), sep = "")))) + -->
<!--   labs(y = "Percent of Genes", -->
<!--        fill = expression(paste("Model and ", italic("Mixta"), " species", sep = ""))) + -->
<!--   theme(axis.text.x = element_text(angle = 45, hjust = 1, face = "italic"),               # Rotates and italicizes x axis labels -->
<!--         legend.position = c(0.95, 0.95),                                                  # Moves legend inside the plot and in the top-right corner -->
<!--         legend.justification = c("right", "top"),                                         # Top right corner is at above coordinates -->
<!--         legend.text.align = 0,                                                            # Aligns legend text to the left -->
<!--         legend.margin = margin(6, 6, 6, 6),                                               # Margins around legend -->
<!--         text = element_text(size = 12,  family = "Times New Roman")) -->
<!-- ``` -->

<!-- ```{r Figure3, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="Figure 3. Closest relatives to *M. calida* genes for nucleotide sequences around the *M. calida* genome. Length and colour of the bars represent the different species to which the gene is most closely related according to genetic distance. The *M. calida* genome has 4084 genes. No distinct patterns emerged; however, genes most closely related to *P. septica* are the most common and occur throughout the genome. Therefore, *P. septica* and *Mixta* are likely to have a recent common ancestor.", fig.width=6.5, fig.height=5} -->
<!-- FR_nucl_calida <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure3_Circular_NT.csv", stringsAsFactors = FALSE) %>% -->
<!--   mutate(Closest_Relative = factor(Closest_Relative, -->
<!--                                    levels = c("P_septica", "P_agglomerans", "E_tasmaniensis", "E_amylovora", "T_ptyseos", -->
<!--                                               "T_saanichensis", "E_cloacae", "P_syringae"), ordered = TRUE)) -->

<!-- ggplot(FR_nucl_calida, aes(x = ID, y = Relative_Number, fill = Closest_Relative)) + -->
<!--   geom_bar(stat = "identity", position = position_dodge(), width = 15) + -->
<!--   coord_polar() + -->
<!--   scale_fill_brewer(palette = "Paired", -->
<!--                     labels = c("P. septica", "P. agglomerans", "E. tasmaniensis", "E. amylovora", "T. ptyseos", "T. saanichensis", "E. cloacae", -->
<!--                                "P. syringae"), -->
<!--                     na.value = "#ff7f00") + -->
<!--   scale_y_continuous(limits = c(0, 9), breaks = c(0, 2, 4, 6, 8)) + -->
<!--   labs(fill = "Closest Relative") + -->
<!--   theme_bw() + -->
<!--   theme(legend.text = element_text(face = "italic"), -->
<!--         axis.title = element_blank(), -->
<!--         axis.text.y = element_blank(), -->
<!--         axis.ticks.y = element_blank(), -->
<!--         text = element_text(size = 12,  family = "Times New Roman")) -->
<!-- ``` -->

<!-- ```{r Figure4, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="Figure 4. Closest relatives to *M. calida* genes for amino acid sequences around the *M. calida* genome. Length and colour of the bars represent the different species to which the gene is most closely related according to genetic distance. The *M. calida* genome has 4084 genes. No distinct patterns emerged; however, genes most closely related to *P. septica* are the most common and occur throughout the genome. Therefore, *P. septica* and *Mixta* are likely to have a recent common ancestor.", fig.width=6.5, fig.height=5} -->
<!-- FR_aa_calida <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure4_Circular_AA.csv", stringsAsFactors = FALSE) %>% -->
<!--   mutate(Closest_Relative = factor(Closest_Relative, -->
<!--                                    levels = c("P_septica", "P_agglomerans", "E_tasmaniensis", "E_amylovora", "T_ptyseos", -->
<!--                                               "T_saanichensis", "E_cloacae", "P_syringae"), ordered = TRUE)) -->

<!-- ggplot(FR_aa_calida, aes(x = ID, y = Relative_Number, fill = Closest_Relative)) + -->
<!--   geom_bar(stat = "identity", position = position_dodge(), width = 15) + -->
<!--   coord_polar() + -->
<!--   scale_fill_brewer(palette = "Paired", -->
<!--                     labels = c("P. septica", "P. agglomerans", "E. tasmaniensis", "E. amylovora", "T. ptyseos", "T. saanichensis", "E. cloacae", -->
<!--                                "P. syringae"), -->
<!--                     na.value = "#ff7f00") + -->
<!--   scale_y_continuous(limits = c(0, 9), breaks = c(0, 2, 4, 6, 8)) + -->
<!--   labs(fill = "Closest Relative") + -->
<!--   theme_bw() + -->
<!--   theme(legend.text = element_text(face = "italic"), -->
<!--         axis.title = element_blank(), -->
<!--         axis.text.y = element_blank(), -->
<!--         axis.ticks.y = element_blank(), -->
<!--         text = element_text(size = 12,  family = "Times New Roman")) -->
<!-- ``` -->

<!-- ```{r Figure5, echo = FALSE, message = FALSE, warning=FALSE, fig.cap="Figure 5. The percentage of genes (out of 799) that were most closely related to any of the eight non-*Mixta* species using percent site identities. The species with the highest percent identity to each of the *Mixta* species is considered the most similar taxon for a single gene. Blue bars represent the nucleotide sequences' results; the light blue bars represent *M. calida* sequences, and the dark blue bars represent *M. gaviniae* sequences. Green bars represent the amino acid sequences' results; the light green bars represent *M. calida* sequences, and the dark green bars represent *M. gaviniae* sequences.", fig.width=6.5} -->
<!-- number_relatives <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure5_FirstRelativeMixtaNTAAIdentity.csv", -->
<!--                              stringsAsFactors = FALSE) %>% -->
<!--   mutate(Species = factor(Species, levels = c("P. septica", "P. agglomerans", "E. tasmaniensis", "E. amylovora", "T. ptyseos", "T. saanichensis", -->
<!--                                               "E. cloacae", "P. syringae"), ordered = TRUE), -->
<!--          Type = factor(Type, levels = c("calida_NT", "gaviniae_NT", "calida_AA", "gaviniae_AA"), ordered = TRUE)) -->


<!-- ggplot(data = number_relatives, aes(x = Species, y = Percent, fill = Type)) + -->
<!--   geom_bar(stat = "identity", position = position_dodge()) + -->
<!--   scale_fill_brewer(palette = "Paired", -->
<!--                     labels = c(expression(paste("NT and ", italic("M. calida"), sep = "")), -->
<!--                                expression(paste("NT and ", italic("M. gaviniae"), sep = "")), -->
<!--                                expression(paste("AA and ", italic("M. calida"), sep = "")), -->
<!--                                expression(paste("AA and ", italic("M. gaviniae"), sep = "")))) + -->
<!--   labs(y = "Percent of Genes", -->
<!--        fill = expression(paste("Sequence type and ", italic("Mixta"), " species", sep = ""))) + -->
<!--   theme(axis.text.x = element_text(angle = 45, hjust = 1, face = "italic"),               # Rotates and italicizes x axis labels -->
<!--         legend.position = c(0.95, 0.95),                                                  # Moves legend inside the plot and in the top-right corner -->
<!--         legend.justification = c("right", "top"),                                         # Top right corner is at above coordinates -->
<!--         legend.text.align = 0,                                                            # Aligns legend text to the left -->
<!--         legend.margin = margin(6, 6, 6, 6),                                               # Margins around legend -->
<!--         text = element_text(size = 12,  family = "Times New Roman")) -->
<!-- ``` -->

<!-- ```{r Figure6, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="Figure 6. The most similar taxa to *M. calida* genes for nucleotide sequences around its genome. The most similar taxon is that which sequence is most identical to the *M. calida* sequence. Length and colour of the bars represent the different species to which the gene is most similar.", fig.width=6.5, fig.height=5} -->
<!-- FI_calida_NT <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure6_CircularIdentities_NT.csv", -->
<!--                          stringsAsFactors = FALSE) %>% -->
<!--   mutate(Closest_Relative = factor(Closest_Relative, -->
<!--                                    levels = c("P_septica", "P_agglomerans", "E_tasmaniensis", "E_amylovora", "T_ptyseos", -->
<!--                                               "T_saanichensis", "E_cloacae", "P_syringiae"), ordered = TRUE)) -->

<!-- ggplot(FI_calida_NT, aes(x = ID, y = Relative_Number, fill = Closest_Relative)) + -->
<!--   geom_bar(stat = "identity", position = position_dodge(), width = 15) + -->
<!--   coord_polar() + -->
<!--   scale_fill_brewer(palette = "Paired", -->
<!--                     labels = c("P. septica", "P. agglomerans", "E. tasmaniensis", "E. amylovora", "T. ptyseos", "T. saanichensis", "E. cloacae", -->
<!--                                "P. syringae")) + -->
<!--   scale_y_continuous(limits = c(0, 9), breaks = c(0, 2, 4, 6, 8)) + -->
<!--   labs(fill = "Most Similar Taxa") + -->
<!--   theme_bw() + -->
<!--   theme(legend.text = element_text(face = "italic"), -->
<!--         axis.title = element_blank(), -->
<!--         axis.text.y = element_blank(), -->
<!--         axis.ticks.y = element_blank(), -->
<!--         text = element_text(size = 12,  family = "Times New Roman")) -->
<!-- ``` -->

<!-- ```{r Figure7, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="Figure 7. The most similar taxa to *M. calida* genes for amino acid sequences around its genome. The most similar taxon is that which sequence is most identical to the *M. calida* sequence. Length and colour of the bars represent the different species to which the gene is most similar.", fig.width=6.5, fig.height=5} -->
<!-- FI_calida_AA <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure7_CircularIdentities_AA.csv", -->
<!--                          stringsAsFactors = FALSE) %>% -->
<!--   mutate(Closest_Relative = factor(Closest_Relative, -->
<!--                                    levels = c("P_septica", "P_agglomerans", "E_tasmaniensis", "E_amylovora", "T_ptyseos", -->
<!--                                               "T_saanichensis", "E_cloacae", "P_syringiae"), ordered = TRUE)) -->

<!-- ggplot(FI_calida_AA, aes(x = ID, y = Relative_Number, fill = Closest_Relative)) + -->
<!--   geom_bar(stat = "identity", position = position_dodge(), width = 15) + -->
<!--   coord_polar() + -->
<!--   scale_fill_brewer(palette = "Paired", -->
<!--                     labels = c("P. septica", "P. agglomerans", "E. tasmaniensis", "E. amylovora", "T. ptyseos", "T. saanichensis", "E. cloacae", -->
<!--                                "P. syringae")) + -->
<!--   scale_y_continuous(limits = c(0, 9), breaks = c(0, 2, 4, 6, 8)) + -->
<!--   labs(fill = "Most Similar Taxa") + -->
<!--   theme_bw() + -->
<!--   theme(legend.text = element_text(face = "italic"), -->
<!--         axis.title = element_blank(), -->
<!--         axis.text.y = element_blank(), -->
<!--         axis.ticks.y = element_blank(), -->
<!--         text = element_text(size = 12,  family = "Times New Roman")) -->
<!-- ``` -->

<!-- ```{r Table5, echo = FALSE, message = FALSE, warning=FALSE, fig.cap=TRUE} -->
<!-- stats_on_all <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/FigureO_AllStatsCalida.csv", stringsAsFactors = FALSE) -->

<!-- same_spp <- data.frame(table(stats_on_all$Same_Species)) -->
<!-- same_spp <- same_spp[which(same_spp$Var1 != "Nope"), ] -->

<!-- same_spp <- rbind(same_spp, data.frame(Var1 = "T_ptyseos", Freq = 0)) -->

<!-- colnames(same_spp) <- c("Species", "Number of Genes") -->

<!-- same_spp <- mutate(same_spp, -->
<!--                    Species = case_when(Species == "E_amylovora" ~ "E. amylovora", -->
<!--                                        Species == "E_tasmaniensis" ~ "E. tasmaniensis", -->
<!--                                        Species == "P_agglomerans" ~ "P. agglomerans", -->
<!--                                        Species == "P_septica" ~ "P. septica", -->
<!--                                        Species == "T_ptyseos" ~ "T. ptyseos", -->
<!--                                        Species == "T_saanichensis" ~ "T. saanichensis", -->
<!--                                        Species == "E_cloacae" ~ "E. cloacae")) %>% -->
<!--   mutate(Species = factor(Species, levels = c("P. agglomerans", "P. septica", "E. amylovora", "E. tasmaniensis", -->
<!--                                               "T. ptyseos", "T. saanichensis", "E. cloacae"), ordered = TRUE)) -->


<!-- same_spp <- arrange(same_spp, Species) -->

<!-- kable(same_spp, caption = "Table 5. The number of genes that had each species as the closest or most similar relative for all four analyses.") -->

<!-- # table(stats_on_all$Same_Genus) -->
<!-- # table(stats_on_all$NT_Same) -->
<!-- # table(stats_on_all$AA_Same) -->
<!-- # table(stats_on_all$Rel_Same) -->
<!-- # table(stats_on_all$Ide_Same) -->
<!-- ``` -->

<!-- ```{r Figure8, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="Figure 8. Sample for first 100 genes.", fig.width=6.5, fig.height=3} -->
<!-- all_results_calida <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/FigureN_AllResultsCalida.csv", -->
<!--                                stringsAsFactors = TRUE) %>% -->
<!--   pivot_longer(cols = c(Rel_NT:Ide_AA), names_to = "Analysis", values_to = "Closest_Similar") %>% -->
<!--   mutate(Closest_Similar = factor(Closest_Similar, -->
<!--                                   levels = c("P_septica", "P_agglomerans", "E_tasmaniensis", "E_amylovora", "T_ptyseos", "T_saanichensis", -->
<!--                                              "E_cloacae", "P_syringiae"), -->
<!--                                   ordered = TRUE), -->
<!--          Analysis = factor(Analysis, levels = c("Ide_AA", "Rel_AA", "Ide_NT", "Rel_NT"), ordered = TRUE), -->
<!--          Gene_Name = reorder(Gene_Name, ID)) -->

<!-- ggplot(all_results_calida[1:400, ], aes(x = Gene_Name, y = Analysis, fill = Closest_Similar)) + -->
<!--   geom_tile() + -->
<!--   scale_fill_brewer(palette = "Paired", -->
<!--                     labels = c("P. septica", "P. agglomerans", "E. tasmaniensis", "E. amylovora", "T. ptyseos", "T. saanichensis", "E. cloacae", -->
<!--                                "P. syringae")) + -->
<!--   scale_y_discrete(labels = c("AA Identity", "AA Closest", "NT Identity", "NT Closest")) + -->
<!--   labs(fill = "Closest Relative or \n Most Similar Taxa") + -->
<!--   theme(legend.text = element_text(face = "italic"), -->
<!--         # legend.position = "bottom", -->
<!--         text = element_text(size = 12,  family = "Times New Roman"), -->
<!--         axis.text.x = element_text(angle = 45, hjust = 1))               # Rotates x axis labels -->

<!-- stats_on_all <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/FigureO_AllStatsCalida.csv", stringsAsFactors = FALSE) -->

<!-- # table(stats_on_all$Same_Species) -->
<!-- # table(stats_on_all$Same_Genus) -->
<!-- # table(stats_on_all$NT_Same) -->
<!-- # table(stats_on_all$AA_Same) -->
<!-- # table(stats_on_all$Rel_Same) -->
<!-- # table(stats_on_all$Ide_Same) -->
<!-- ``` -->

<!-- ```{r Figure9, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="Figure 9. Maximum likelihood gene trees for ttuB for nucleotide (A) and amino acid sequences (B).", fig.width=3.25} -->
<!-- knitr::include_graphics("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure9_ttuB_NT.png") -->
<!-- knitr::include_graphics("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure9_ttuB_AA.png") -->
<!-- ``` -->

<!-- ```{r Figure10, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="Figure 10. Maximum likelihood gene trees for secY for nucleotide (A) and amino acid sequences (B). Since bifucations could not be seen in radial trees, condensed trees were computed with a 70% cutoff for both the nucleotide sequnces (C) and amino acid sequences (D).", fig.width=3.25} -->
<!-- knitr::include_graphics("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure10_secY_NT.png") -->
<!-- knitr::include_graphics("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure10_secY_AA.png") -->
<!-- knitr::include_graphics("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure10b_secY_NT70.png") -->
<!-- knitr::include_graphics("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure10b_secY_AA70.png") -->
<!-- ``` -->

# Discussion

# Acknowledgements

# Contributions to the Field Statement

# Figure Legends

# Tables


\newpage
# References
<div id="refs"></div>
