---
title: "An investigation into the contentious phylogenetic results of the *Mixta* species"
author: "Kimberly Hinz"
date: "2020/12/08"
output:
  word_document:
    reference_docx: Styles1.docx
    fig_caption: true
    # toc: true
    # toc_depth: 5
bibliography: References.bib
csl: IJSEM.csl
---
```{r Packages, echo = FALSE, message = FALSE, warning=FALSE}
library("knitr")
library("tidyverse")
library("ggplot2")
library("RColorBrewer")
library("extrafont")
theme_set(theme_classic())
```
\newpage
# Abstract
*Mixta* is a recently described genus within the bacterial family *Erwiniaceae* and is closely related to *Pantoea*, *Erwinia*, and *Tatumella*. Depending on the genes and method used for phylogenetic analysis, the genus with which *Mixta* shares the most recent ancestor differs. This study aimed to determine the cause behind these contentious results and identified genes that may have resulted from horizontal gene transfer. Ten complete or partial genomes – two *Mixta* species, six species from other *Erwiniaceae* genera, and two outgroups – were retrieved from NCBI and annotated. Homologous genes were extracted yielding a dataset of 799 genes. The genes were aligned with the ClustalW algorithm, and `MEGA-CC` was used for model testing and computing phylogenetic trees for each gene for both nucleotide and amino acid sequences. Distance matrices were then extracted from these trees. Nucleotide and amino acid identity analyses were also done using the programming language `R`. *Pantoea* was the closest relative to the *Mixta* species in most analyses; however, results were not consistent. Some genes were also found to be more similar to other, non-*Pantoea* species. Diligence must be given to selecting genes for phylogenetic analysis and the method chosen to prevent any xenologous signal from distorting the actual relationships. Furthermore, future research should consider that different phylogenetic analyses may provide different results.

\newpage
# Introduction
The bacterial family Erwiniaceae includes diverse Gram-negative [@Winslow1917; @Prakash2015] organisms, including the genera *Mixta*, *Pantoea*, *Erwinia*, *Tatumella*, *Buchnera*, *Wigglesworthia*, *Phaseolibacter*, *Kalamiella* (LPSN, http://www.bacterio.net/). Numerous studies have focused on three of these genera – *Mixta*, *Pantoea*, and *Erwinia* – given their relevance to plant and human pathogens. For example, the genus *Erwinia* contains plant pathogens (e.g. *E. amylovora*) [@HaubenSwings2005; @Lopez2011; @Matsuura2012], epiphytes (e.g. *E. tasmaniensis*) [@Geider2006; @Mergaert1999], species that may contribute to plant disease [@Buonaurio2015], and others that are insect-associated [@Campillo2015; @Harada1997; @Skrodenyte-Arbaciauskiene2012]. Similarly, *Pantoea*, a genus reported by Gavini et al.[@Gavini1989], includes species that inhabit a variety of environments [@Gavini1989; @Popp2010; @Liu2013], including human and plant pathogens [@Prakash2015; @deBaere2004; @Fullerton2007; @Kratz2003; @Lim2006; @Schmid2003; @vanRostenberghe2006] as well as species associated with plant growth [@Smits2011; @Kim2012], insects [@Palmer2016], and fungi [@Ma2016; @Rong2016]. *Mixta*, originally *Pantoea*, is a novel genus first suggested by Palmer et al. [@Palmer2017] and later proposed by the same researchers in 2018 [@Palmer2018]. *M. calida* and *M. gaviniae* (previously species of *Pantoea*) were initially found in infant formula and its production environment [@Popp2010]. Fritz et al. [@Fritz2014] demonstrated that some strains of *M. calida* were linked with post-surgical meningitis in humans. Currently, there are only five *Mixta* species validly published (LPSN, http://www.bacterio.net/).

Understanding the phylogeny of bacterial species and genes allows researchers to investigate the species' evolutionary histories, relatedness, and the individual genes that enable them to survive under various conditions [@Palmer2017; @Bennett2012; @Prasanna2013; @Angus2014; @deMaayer2014; @Fouts2016]. The most common methods in phylogenetic analysis are those that use ubiquitous genes including 16S ribosomal RNA (rRNA) sequences [@Konstantinidis2007], multilocus sequence analysis (MLSA) with several housekeeping genes [@Konstantinidis2007; @Gevers2005; @Glaeser2015], and ribosomal MLSA with 53 structural ribosomal proteins [@Palmer2017; @Bennett2012; @Jolley2012]. However, the phylogenetic trees generated from these methods may not be robust [@Konstantinidis2007; @Glaeser2015; @Brady2008] because of the lower phylogenetic signal from the highly conserved genes [@Konstantinidis2007; @Gevers2005; @Fox1992; @Konstantinidis2005; @Staley2006; @RichterRosselloMora2009]. This issue is particularly the case with *Erwiniaceae* because much of our understanding is based on the 16S rRNA and other housekeeping genes [@Francino2006; @Hauben1998; @Naum2008; @Sproer1999; @Adeolu2016]. However, such analyses have an insufficient resolution at the species level [@Glaeser2015], and the genera *Pantoea*, *Erwinia*, *Tatumella*, and *Mixta* are often not monophyletic [@Rezzonico2016]. Another issue is the unintentional inclusion of paralogues (originating from an intragenomic duplication event) or xenologues (arising from horizontal gene transfer) in MLSA gene analyses [@Palmer2017; @Koonin2005].

The DNA gyrase subunit (gyrB), DNA-directed RNA polymerase subunit (rpoB), ATP synthase subunit (atpD), and initiation translation factor (infB) are more reliable than 16S rRNA gene sequences for determining intra- and intergeneric relationships [@Hedegaard1999; @Dauga2002; @Mollet1997; @Paradis2005] and a robust method for the Erwiniaceae family [@Brady2010]. The genera within Erwiniaceae are monophyletic when the analysis uses concatenated partial gyrB, rpoB, atpD, and infB gene sequences [@Brady2008; @Brady2009; @Moretti2011].

Nevertheless, despite efforts to find a suitable method for identifying the species tree for Erwiniaceae genera, *Mixta*’s position relative to *Pantoea*, *Erwinia*, and *Tatumella* vastly differs between studies [@Prakash2015; @Palmer2017; @Palmer2018; @Adeolu2016; @Brady2011; @Rezzonico2016; @Brady2012; @Gueule2015]. Not only do different gene combinations provide different results, but different algorithms and models for alignments and phylogenetic trees also contribute to the contentious results [@Naum2008; @Octavia2014]. For example, Prakash et al. [@Prakash2015], Palmer et al. [@Palmer2017], Rezzonico et al. [@Rezzonico2016], and Gueule et al. [@Gueule2015] all used gyrB, rpoB, atpD, and infB nucleotide gene sequences in their phylogenetic analyses. Prakash et al. conducted a neighbour-joining statistical method while the others used a maximum-likelihood (ML) method. Three of the studies – Rezzonico et al., Prakash et al., and Gueule et al. – placed the *Mixta* species within the *Pantoea* genera. In contrast. the analysis in Palmer et al. identified *Mixta* outside of the other three genera. Palmer et al. [@Palmer2018], Rezzonico et al. [@Rezzonico2016], and Brady et al. [@Brady2012] also ran analyses with the four MLSA genes as amino acid sequences. *Mixta* was found to be related closest to *Tatumella* by Rezzonico et al. and *Pantoea* by Palmer et al. Brady et al. suggested that *Mixta* was a part of the *Pantoea* genus and not separate. A separate study by Palmer et al. @Palmer2017 computed two ML trees; one used 1039 core amino acid sequences, and the other used 52 ribosomal MLSA nucleotide sequences. The 1039 core gene tree suggested *Pantoea* and *Tatumella* were sister taxa followed by two *Mixta* species. However, the ribosomal MLSA gene tree proposed *Pantoea* and *Tatumella* were sister taxa followed by *Erwinia* and then by *Mixta*.

Two possible causes may be behind the contentious results in phylogenetic analyses. The first is the methods; this includes the software and the applied model; two different phylogenetic tree models will provide different results even if both trees are ML and use the same genes and sequence type. The second potential cause is the materials: the gene sequences. The inclusion of different genes between analyses may give different results if one or more in the group is xenologous or has a different evolutionary rate. Xenologous genes need to be identified and removed from the dataset before phylogenetic analysis of gene sets since horizontal gene transfer (HGT) events frequently occur within bacteria and may occur between members of the same family. Sequence type may also affect results due to codon redundancy and codon usage bias. Additionally, there is no common phylogenetic tree model between nucleotide and amino acid sequences; therefore, evolutionary history is estimated differently between sequence types.

This paper aims to determine the cause behind the *Mixta* species' contentious phylogenetic results. Ten complete or partial genomes – two for each of *Mixta*, *Pantoea*, *Erwinia*, and *Tatumella* and two outgroups – were retrieved from a public database. Homologous genes were extracted from these genomes in both the nucleotide and amino acid sequences. Distance matrices were exported from the phylogenetic trees, and these were used to determine which species were most closely related to *Mixta* for each gene. Nucleotide and amino acid identity analyses were also conducted for each gene to compare the results between evolutionary history and sequence similarity. Phylogenetic trees were then done with either one gene or multiple genes to confirm results.

# Materials and Methods
## Genomes and Gene Sets
Ten publicly available full or partial genomes of the species' type strains were retrieved from the National Center for Biotechnology Information (NCBI). Each of the four *Erwiniaceae* genera had two representatives, and *Enterobacter cloacae* and *Pseudomonas syringae* are the outgroups (Table 1). *E. cloacae* belongs to the same order as *Erwiniacae* (*Enterobacterales*), and *P. syringae* belongs to the same class (*Gammaproteobacteria*). 

The genomes were annotated with Prokka v1.14.1 [@Seemann2014] using default parameters, and homologous genes were extracted using GET_HOMOLOGUES software package [@Contreras-Moreira2013] with the bidirectional best-hit search algorithm using default parameters. GET_HOMOLOGUES extracted 954 homologous genes between the ten species.

```{r Table1, echo=FALSE, message=FALSE, warning=FALSE, fig.cap=TRUE}
strain <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Table1_Genomes.csv", stringsAsFactors = FALSE)

kable(strain, caption = "Table 1. Genomes used in this study.")
```

In `R` [@RCore2019], two filter functions were written to pass through the nucleotide sequence gene files. Since both nucleotide and amino acid sequences were analyzed, the filters were only applied to the nucleotide sequences. A list of the passed nucleotide gene files was used to isolate the corresponding amino acid gene files. The first filter kept genes files that contained ten, single-copy sequences that were approximately the same length. The shortest gene length allowed was 90% the length of the longest sequence in the gene file. For example, if the longest sequence in a gene file was 1000 bp, then the other sequences had to be at least 900 bp. Consequently, 741 gene files passed through this filter. Excluded from the dataset were 38 gene files because at least one species had a second copy and another 175 were excluded for not meeting the length requirements.

Since there may be an evolutionary reason why at least one of the sequences did not meet the length requires, a second filter was applied to the 175 gene files that did not meet the initial length requirements. This filter took each file at a time and separated the sequences into two groups. One group contained the sequences that were longer than the initial 90% length cutoff and the other contained the sequences between 80 and 90% the length of the longest gene in the file. If at least one of the sequences was below this new 80% cutoff, then the file was excluded. To pass through, both representatives of a genus had to be in the same group. An additional 58 genes were added to the working dataset, resulting in 799 genes. A list of the 799 gene names was used to grab the corresponding amino acid files.

## Gene Alignment
Gene alignments were in `R` using a ClustalW algorithm. The gene fasta files were read into `R` using the `readDNAStringSet()` function (package: Biostrings) [@Pages2020]. Alignments were done using the `msaClustalW()` function (package: msa) [@Bodenhofer2015] using default parameters and setting the maximum number of iterations to 100. The aligned sequences were then converted into a readable format using `msaConvert()` (package: msa)[@Bodenhofer2015] and written into a new fasta file using the `write.fasta()` function (package: seqinr) [@Charif2007]

## Distance Matrices
After alignments, model testing was done for each gene through `MEGA-CC` [@Kumar2012] for both nucleotide and amino acid sequences. `MEGA-CC` is called in a command-line terminal along with a text file listing the file pathways to all 799 genes and a MAO file detailing the analysis preferences. Default parameters were used for model testing. The output was a comma-separated values (CSV) file for each gene. These CSV files were read into `R`, and the model for phylogenetic analysis was extracted based on the lowest Bayesian Information Criterion (BIC). Twelve and twenty-three unique models were used for nucleotide (Table 2) and amino acid sequences (Table 3) were needed to create phylogenetic gene trees.

```{r Table2, echo=FALSE, message=FALSE, warning=FALSE, fig.cap=TRUE}
PM_NT <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Table2_PmodelsNT.csv", stringsAsFactors = FALSE)

kable(PM_NT, caption = "Table 2. According to model testing and the lowest BIC for nucleotide sequences, the number of genes required each phylogenetic tree model.")
```

```{r Table3, echo = FALSE, message = FALSE, warning=FALSE, fig.cap=TRUE}
PM_AA <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Table3_PmodelsAA.csv", stringsAsFactors = FALSE)

kable(PM_AA, caption = "Table 3. According to model testing and the lowest BIC for amino acid sequences, the number of genes required each phylogenetic tree model.")
```

The gene file paths were grouped according to unique models. For each model, a text file was written containing the pathways. `MEGA-CC` [@Kumar2012] was called in a command-line terminal along with a file pathways text file and the corresponding MAO file detailing the unique model. This step was repeated for each model. The MAO files specified the use of a maximum likelihood statistical method, 500 bootstrap replications, the model, and rate patterns to create a phylogenetic tree for each gene. If a model required a Gamma distribution, the number of discrete Gamma categories was set to 5, which is the default.

Distance matrices were extracted from each of the phylogenetic trees and saved as Excel files. These files were then read into `R`. The genetic distances between *Mixta* and the other species were extracted for further analysis in `R`. Tables were created in `R Markdown` using the `kable()` function (package: knitr) [@Xie2020] and figures were created using the `ggplot()` function (package: ggplot2) [@Wickham2016].

## Nucleotide and Amino Acid Identity
Individual species sequences of genes were aligned to both of the corresponding *Mixta* sequences in turn. Alignments were done in `R` using the same methods as for gene alignment of the whole gene file. The aligned sequences were then split into the individual sites using the `str_split()` function (package: stringr) [@Wickham2019]. A function was created using base `R` code to calculate the number of identical sites between the two sequences over the total number of sites to give a similar identity percentage. Values that equal or are close to 100% are more identical to the *Mixta* sequences.

# Results and Discussion
After the two filters, 799 of 954 homologous genes remained. The genes included those encoding rRNA sequences, enzymes, transporters, and housekeeping genes and were scattered across the *M. calida* and *M. gaviniae* genomes.

## Genetic Distance
### Closest Relative from Genetic Distances
Genetic distance gives a numeric estimate of evolutionary change between species. The species that has the shortest genetic distance to a species of interest is considered the closest relative. For the following analysis, model testing was done first to estimate the best model for each gene. Next, individual phylogenetic gene trees were created, and it was from these that distance matrices were extracted. *P. septica* was the closest relative to 66.3% and 66.5% of the *M. calida* and *M. gaviniae* genes, respectively, for the nucleotide sequences (Figure 1). In comparison, only 42.7% and 42.8% of *M. calida* and *M. gaviniae* genes, respectively, are closely related to *P. septica* for the amino acid sequences (Figure 1), but this is still the majority. These results suggest that *P. septica* is the closest relative to *Mixta*, followed by *P. agglomerans*. In contrast, the *Mixta* genes are more closely related to the other species in the study than *P. septica* in the amino acid analysis (Figure 1). For instance, the percentage of genes related to *P. agglomerans* increased by 8.4 and 8.5% for *M. calida* and *M. gaviniae*, respectively (Figure 1). This may be due to a loss of phylogenetic signal when using amino acid sequences caused by codon degeneracy and having a third of available sites.

```{r Figure1, echo = FALSE, message = FALSE, warning=FALSE, fig.cap="Figure 1. The percentage of genes (out of 799) that were most closely related to any of the eight non-*Mixta* species using genetic distances. The species with the shortest genetic distance from both of the *Mixta* species is considered the closest relative. Blue bars are the nucleotide sequences of the genes with light blue bars representing the percentage of *M. calida* genes and dark blue bars representing *M. gaviniae*. Green bars are the corresponding amino acid sequences with light green representing *M. calida* genes and dark blue bars representing *M. gaviniae*. Approximately 66% of *M. calida* and *M. gaviniae* genes were most closely related to *P. septica* when using nucleotide sequences compared to only 43% when using amino acid sequences.", fig.width=6.5}
number_relatives <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure1_FirstRelativeMixtaNTAA.csv", 
                             stringsAsFactors = FALSE) %>%
  mutate(Species = factor(Species, levels = c("P. septica", "P. agglomerans", "E. tasmaniensis", "E. amylovora", "T. ptyseos", "T. saanichensis", 
                                              "E. cloacae", "P. syringae"), ordered = TRUE),
         Type = factor(Type, levels = c("calida_NT", "gaviniae_NT", "calida_AA", "gaviniae_AA"), ordered = TRUE))


ggplot(data = number_relatives, aes(x = Species, y = Percent, fill = Type)) +
  geom_bar(stat = "identity", position = position_dodge()) +
  scale_fill_brewer(palette = "Paired", 
                    labels = c(expression(paste("NT and ", italic("M. calida"), sep = "")), 
                               expression(paste("NT and ", italic("M. gaviniae"), sep = "")), 
                               expression(paste("AA and ", italic("M. calida"), sep = "")), 
                               expression(paste("AA and ", italic("M. gaviniae"), sep = "")))) +
  labs(y = "Percent of Genes", 
       fill = expression(paste("Sequence type and ", italic("Mixta"), " species", sep = ""))) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, face = "italic"),               # Rotates and italicizes x axis labels
        legend.position = c(0.95, 0.95),                                                  # Moves legend inside the plot and in the top-right corner
        legend.justification = c("right", "top"),                                         # Top right corner is at above coordinates
        legend.text.align = 0,                                                            # Aligns legend text to the left
        legend.margin = margin(6, 6, 6, 6),                                               # Margins around legend
        text = element_text(size = 12,  family = "Times New Roman"))
```

Unexpectedly, the number of *Mixta* genes that were more closely related to *E. cloacae* increased by 3.1% when amino acid sequences were used (Figure 1). The differences in relatedness for *Mixta* genes between nucleotide and amino acid sequences suggests that sequence type impacts the results, even when using the same genes.

If genetic distance infers the same closest relative for a *Mixta* gene for both nucleotide and amino acid analyses, these results may be more reliable. Thus, for genes where the closest relative is an *Erwinia*, *Tatumella*, or *Enterobacter* species both nucleotide and amino acid analyses, they may have resulted from a horizontal gene transfer (HGT) event rather than vertical evolution from a *Pantoea* species. No *Mixta* genes were closely related to *P. syringae* for either the nucleotide or amino acid sequences (Figure 1). This result confirms that *P. syringae* is a suitable outgroup for this analysis and suggests that any HGT events may be restricted to  *Enterobacterales*.

### Best Model vs Best Available Model
In `MEGAX` [@Kumar2018], there are fewer model options for genetic distance than for phylogenetic trees. However, model testing using this software tests all of the available models for phylogenetic tree analysis. Therefore, the recommended model based on the lowest BIC value may not be available for genetic distance analysis. GTR (General Time Reversible) and HKY (Hasegawa-Kishino-Yano) models, as well as the invariant sites (+I) parameter, are not available for genetic distance analysis. Genetic distances for the nucleotide sequences were estimated in `MEGA-CC` using the limited model options to determine if this would affect closest relative percentages. The model testing results were filtered to remove the models not available for genetic distances in `MEGAX`. Then distance matrices were estimated for each gene using the best available model based on the lowest BIC value. As a result, only four of twelve models were available with more than 87% of the genes now using either the Tamura 3-parameter with a Gamma parameter (T92_G) or the Tamura-Nei with a Gamma parameter (TN93_G) (Table 4). When not restricted by availability, only 58% of the genes required these models (Table 4). Furthermore, the best model was not available for approximately 30.3% of the genes (242 genes). The mean BIC difference between the best and best available models  was 2.65 (range: 0, 122.6). In this instance, 0 means that the best available model is the same as the best model. When these zeroes are ignored, the mean BIC difference was 13.8 (range: 0.1, 122.6). These differences may result in a less accurate estimation of genetic distance between species.

```{r Table4, echo = FALSE, message = FALSE, warning=FALSE, fig.cap=TRUE}
models <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Table4_GenDistModelsNT.csv", stringsAsFactors = FALSE)

colnames(models) <- c("Model", "Number of genes", "Number of genes, if available")

kable(models, caption = "Table 4. The number of genes requiring each model when using phylogenetic tree analysis vs genetic distance analysis in MEGAX for nucleotide sequences.")
```

The closest relatives to both *Mixta* species were extracted from the distance matrices. *P. septica* was estimated to be the closest relative to 12.6 and 14.4% more genes for *M. calida* and *M. gaviniae*, respectively, when the best available models were used (Figure 2). The next most considerable difference was for genes calculated to be most closely related to *P. agglomerans* (Figure 2). Approximately 5.5 and 7.1% fewer genes were most closely related to *P. agglomerans* when the best available model was used (Figure 2). Additionally, there was more disparity in the closest relatives between *M. calida* and *M. gaviniae* when only the best available model was used (Figure 2). When the best model was used, the two *Mixta* typically had the same closest relative for each gene; however, this was not the case when model options were limited to those available. Due to differences in evolutionary change estimation between models and the addition or exclusion of additional parameters, genetic distances may be inaccurately calculated when the best model is not used. Such results may even suggest that sibling taxa are not as closely related as initially hypothesized.

```{r Figure2, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="Figure 2. The percentage of genes (out of 799) that were most closely related to any of the eight non-*Mixta* species when using the recommended (best) model from model testing vs the restricted model (best available) options available for genetic distance analysis in MEGAX for nucleotide sequences. The species with the shortest genetic distance from both of the *Mixta* species is considered the closest relative. Evolutionary patterns may vary by gene; therefore, it is appropriate to conduct model testing and use the model that explains the most variation. Nevertheless, this model may not be available in software programs or analysis methods within a software program. Pink bars represent the percentage of *Mixta* genes most closely related to non-*Mixta* species when using the recommended model. Orange bars represent the percentage of *Mixta* genes most closely related to any non-*Mixta* species when using the best available genetic distance model analysis in MEGAX. GTR and HKY models and invariant sites parameter were not available for this analysis. The lighter shades represent *M. calida*, and the darker shades represent *M. gaviniae*.", fig.width=6.5}
number_relatives <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure2_FirstRelativeBestvsAvailableModelNT.csv", 
                             stringsAsFactors = FALSE) %>%
  mutate(Species = factor(Species, levels = c("P. septica", "P. agglomerans", "E. tasmaniensis", "E. amylovora", "T. ptyseos", "T. saanichensis", 
                                              "E. cloacae", "P. syringae"), ordered = TRUE),
         Type = factor(Type, levels = c("PD_calida", "PD_gaviniae", "GD_calida", "GD_gaviniae"), ordered = TRUE))

ggplot(data = number_relatives, aes(x = Species, y = Percent, fill = Type)) +
  geom_bar(stat = "identity", position = position_dodge()) +
  scale_fill_manual(values = c("#fb9a99", "#e31a1c", "#fdbf6f", "#ff7f00"),
                    labels = c(expression(paste("Best Model and ", italic("M. calida"), sep = "")), 
                               expression(paste("Best Model and ", italic("M. gaviniae"), sep = "")), 
                               expression(paste("Best Available Model and ", italic("M. calida"), sep = "")), 
                               expression(paste("Best Available Model and ", italic("M. gaviniae"), sep = "")))) +
  labs(y = "Percent of Genes", 
       fill = expression(paste("Model and ", italic("Mixta"), " species", sep = ""))) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, face = "italic"),               # Rotates and italicizes x axis labels
        legend.position = c(0.95, 0.95),                                                  # Moves legend inside the plot and in the top-right corner
        legend.justification = c("right", "top"),                                         # Top right corner is at above coordinates
        legend.text.align = 0,                                                            # Aligns legend text to the left
        legend.margin = margin(6, 6, 6, 6),                                               # Margins around legend
        text = element_text(size = 12,  family = "Times New Roman"))
```

About 20.0 and 22.3% of *M. calida* and *M. gaviniae* genes, respectively, had a different closest relative between the best and the best available models. Thus, of the 30.3% of genes with different models between the two analyses, 10.3% of the *M. calida* genes and 8.0% of the *M. gaviniae* genes had the same closest relative. These genes' results may be the same because the two models were similar and explained approximately the same amount of variance. Genetic distances between the species may also be sufficiently large that it does not change which species is most closely related to *Mixta*.

Each model estimates genetic distance differently. The best model explains most of the variance between sequences and gives a more accurate estimation of genetic distance. Using a model that does not explain most of the variance will estimate genetic distance less accurately, resulting in related taxa (such as those from the same species) having a longer genetic distance between them than in actuality. This may also result in the opposite for taxa that are not closely related (i.e. *Pantoea* spp. and *E. cloacae*). Even though only two nucleotide models and one parameter were unavailable for genetic distance analysis in `MEGAX`, the results for the closest relative to *Mixta* genes were substantially impacted. This was particularly the case between the two *Mixta* as one would expect them to have the same closest relative for most of the homologous genes. Therefore, using the best model is recommended. However, this may not be possible when estimating genetic distance using more than one gene that may require different models. Thus, it must be considered that one or more the genes in the concatenation may have a different evolutionary pattern than the others and results should be investigated further to ensure accuracy.

Although genetic distance estimation in `MEGAX` did not offer all of the models and parameters as suggested by model testing, all of the models and parameters were available for phylogenetic tree analysis and genetic distance can be exported from these trees. However, this has to be done manually by opening the tree file in `MEGAX` and exporting the distance matrix as either an Excel or CSV file. Therefore, it is not feasible when hundreds or thousands of genetic distance matrices are needed.

### Spatial Patterns of Closest Relatives
Circular plots were generated using the *Mixta* gene IDs for nucleotide (Figure 3) and amino acid sequences (Figure 4). These were created to assess whether there were any positional effects of the genes and spatial patterns in the closest relative. The gene IDs give the approximate location of each gene relative to the *Mixta* genome. The length and colour of a bar represent the species that was the closest relative to the *Mixta* gene at that position. Since the closest relatives were the same both *Mixta* species except for 1 and 2 genes for nucleotide and amino acid sequences, respectively, only the results for *M. calida* are shown. These genes will be discussed later on.

```{r Figure3, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="Figure 3. Closest relatives to *M. calida* genes for nucleotide sequences around the *M. calida* genome. Length and colour of the bars represent the different species to which the gene is most closely related according to genetic distance. The *M. calida* genome has 4084 genes. No distinct patterns emerged; however, genes most closely related to *P. septica* are the most common and occur throughout the genome. Therefore, *P. septica* and *Mixta* are likely to have a recent common ancestor.", fig.width=6.5, fig.height=5}
FR_nucl_calida <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure3_Circular_NT.csv", stringsAsFactors = FALSE) %>%
  mutate(Closest_Relative = factor(Closest_Relative, 
                                   levels = c("P_septica", "P_agglomerans", "E_tasmaniensis", "E_amylovora", "T_ptyseos", 
                                              "T_saanichensis", "E_cloacae", "P_syringae"), ordered = TRUE))

ggplot(FR_nucl_calida, aes(x = ID, y = Relative_Number, fill = Closest_Relative)) +
  geom_bar(stat = "identity", position = position_dodge(), width = 15) +
  coord_polar() +
  scale_fill_brewer(palette = "Paired", 
                    labels = c("P. septica", "P. agglomerans", "E. tasmaniensis", "E. amylovora", "T. ptyseos", "T. saanichensis", "E. cloacae",
                               "P. syringae"),
                    na.value = "#ff7f00") +
  scale_y_continuous(limits = c(0, 9), breaks = c(0, 2, 4, 6, 8)) +
  labs(fill = "Closest Relative") +
  theme_bw() +
  theme(legend.text = element_text(face = "italic"),
        axis.title = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        text = element_text(size = 12,  family = "Times New Roman"))
```

For the nucleotide sequences, there were no discernible patterns in closest relatives to *M. calida* genes except for the seven genes most closely related to a *Tatumella* species (pinks) which are only in the first half of the *M. calida* genome (Figure 3). Genes that are most closely related to *Pantoea* species (blue) were distributed throughout the genome (Figure 3), which further suggests that *Mixta* and *Pantoea* have a recent ancestor. Unexpectedly, genes closely related to other genera, particularly to *E. cloacae* (orange) are also distributed throughout the genome (Figure 3). 

Fewer genes were closely related to *P. septica* when analyzing the amino acid sequences (Figure 4). Instead, more genes were closely related to *P. agglomerans*, the *Erwinia* species, the *Tatumella* species, and *E. cloacae* (Figure 4). Nevertheless, no apparent patterns arose in the closest relatives to the *Mixta* genes (Figure 4). The *Mixta* genes that have changed their allegiance between the nucleotide and amino acid analyses may result from the redundancy of the genetic code.

```{r Figure4, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="Figure 4. Closest relatives to *M. calida* genes for amino acid sequences around the *M. calida* genome. Length and colour of the bars represent the different species to which the gene is most closely related according to genetic distance. The *M. calida* genome has 4084 genes. No distinct patterns emerged; however, genes most closely related to *P. septica* are the most common and occur throughout the genome. Therefore, *P. septica* and *Mixta* are likely to have a recent common ancestor.", fig.width=6.5, fig.height=5}
FR_aa_calida <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure4_Circular_AA.csv", stringsAsFactors = FALSE) %>%
  mutate(Closest_Relative = factor(Closest_Relative, 
                                   levels = c("P_septica", "P_agglomerans", "E_tasmaniensis", "E_amylovora", "T_ptyseos", 
                                              "T_saanichensis", "E_cloacae", "P_syringae"), ordered = TRUE))

ggplot(FR_aa_calida, aes(x = ID, y = Relative_Number, fill = Closest_Relative)) +
  geom_bar(stat = "identity", position = position_dodge(), width = 15) +
  coord_polar() +
  scale_fill_brewer(palette = "Paired", 
                    labels = c("P. septica", "P. agglomerans", "E. tasmaniensis", "E. amylovora", "T. ptyseos", "T. saanichensis", "E. cloacae",
                               "P. syringae"),
                    na.value = "#ff7f00") +
  scale_y_continuous(limits = c(0, 9), breaks = c(0, 2, 4, 6, 8)) +
  labs(fill = "Closest Relative") +
  theme_bw() +
  theme(legend.text = element_text(face = "italic"),
        axis.title = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        text = element_text(size = 12,  family = "Times New Roman"))
```

If many genes were horizontally transferred in a single event, then large blocks of genes from a single species would be expected. However, this is not the case for any of species. There are several hypotheses as to why. The first is that may have *Mixta* genome underwent many HGT events after speciation from *Pantoea*; however, further research into the origins of these genes and the frequency of HGT events with *Mixta* and other genera within the *Erwiniaceae* is needed. A second hypothesis is that the gene sequence is so well-conserved that the genetic distance between *Mixta* and more than one of the other species are near identical. Since only one species was extracted for the closest relative to each *Mixta* gene, a species may have been at random if there was a tie between more than one species for having the shortest distance to *Mixta*. For example, suppose the genetic distance from *M. calida* to both *P. septica* and *E. amylovora* for a single gene was 0.012, and this was the shortest distance for that gene. In that case, both species are tied for the closest relative for that gene. However, since only one can be determined as the closest relative, the closest relative may have been decided by the computer program based on another characteristic such as alphabetical order. Therefore, quantitative analysis or computing a phylogenetic tree to view the two-dimensional results is needed.

## Nucleotide and Amino Acid Identity
### Most Similar Taxa from Site Identity
Site identity is here defined as the amount of similarity between two sequences. If two sequences have a site identity of 100%, then they are identical. If site identity is 99%, then there is up to a few bases or amino acids that are different between the two sequences depending on the sequences' length. Within each gene file, the ten sequences were aligned to both *M. calida* and *M. gaviniae* in turn, and the site identity was calculated. Site identity was divided by the length of aligned genes to give a per cent identity that could be compared across the genes and between the nucleotide and amino acid sequences. From these values, the closest non-*Mixta* relative was extracted for each gene.

The majority of the *Mixta* sequences were similar to *P. septica* for nucleotide sequences (Figure 5) with 80.6 and 79.3% of *M. calida* and *M. gaviniae* sequences, respectively. These results are approximately 15% more genes than in the genetic distance analysis. The rest of the genes are most similar to *P. agglomerans*, followed by the *Erwinia* species, *E. cloacae*, and then the *Tatumella* species. Viable mutations that do not disrupt the ability to survive and reproduce are rare and occur at a constant rate. Thus, these results suggest that *P. septica* and the *Mixta* species share a recent common ancestor.

```{r Figure5, echo = FALSE, message = FALSE, warning=FALSE, fig.cap="Figure 5. The percentage of genes (out of 799) that were most closely related to any of the eight non-*Mixta* species using percent site identities. The species with the highest percent identity to each of the *Mixta* species is considered to be most similar taxon for a single gene. Blue bars are the results for the nucleotide sequences where the light blue bars represent *M. calida* sequences and the dark blue bars represent *M. gaviniae* sequences. Green bars are the results for the amino acid sequences where the light green bars represent *M. calida* sequences and the dark green bars represent *M. gaviniae* sequences.", fig.width=6.5}
number_relatives <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure5_FirstRelativeMixtaNTAAIdentity.csv", 
                             stringsAsFactors = FALSE) %>%
  mutate(Species = factor(Species, levels = c("P. septica", "P. agglomerans", "E. tasmaniensis", "E. amylovora", "T. ptyseos", "T. saanichensis", 
                                              "E. cloacae", "P. syringae"), ordered = TRUE),
         Type = factor(Type, levels = c("calida_NT", "gaviniae_NT", "calida_AA", "gaviniae_AA"), ordered = TRUE))


ggplot(data = number_relatives, aes(x = Species, y = Percent, fill = Type)) +
  geom_bar(stat = "identity", position = position_dodge()) +
  scale_fill_brewer(palette = "Paired", 
                    labels = c(expression(paste("NT and ", italic("M. calida"), sep = "")), 
                               expression(paste("NT and ", italic("M. gaviniae"), sep = "")), 
                               expression(paste("AA and ", italic("M. calida"), sep = "")), 
                               expression(paste("AA and ", italic("M. gaviniae"), sep = "")))) +
  labs(y = "Percent of Genes", 
       fill = expression(paste("Sequence type and ", italic("Mixta"), " species", sep = ""))) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, face = "italic"),               # Rotates and italicizes x axis labels
        legend.position = c(0.95, 0.95),                                                  # Moves legend inside the plot and in the top-right corner
        legend.justification = c("right", "top"),                                         # Top right corner is at above coordinates
        legend.text.align = 0,                                                            # Aligns legend text to the left
        legend.margin = margin(6, 6, 6, 6),                                               # Margins around legend
        text = element_text(size = 12,  family = "Times New Roman"))
```

In contrast, only 47.8 and 47.6% of *M. calida* and *M. gaviniae* sequences were most similar to *P. septica* for the amino acid sequences (Figure 5). Instead, the genes are most similar to either *P. agglomerans* or one of the two *Erwinia* species (Figure 5). Additionally, 3.1 and 2.5% more genes were most similar to the *E. cloacae* sequences for *M. calida* and *M. gaviniae*, respectively (Figure 5). In the identity analyses, amino acids that are functionally equivalent were not considered. This may explain the large differences between the nucleotide and amino acid identities for which species was most similar to the *Mixta* sequences. Additionally, at every site for a nucleotide sequence, each base has a 1:3 ratio of being identical to the base at the corresponding location of the other sequence whereas there is only a 1:19 ratio of an amino acid being identical to the amino acid at the same position of the other sequence. Thus, it is far more likely for a nucleotide at a random position to be identical the nucleotide at the same position of the other sequence. This may lead to problems when only amino acid sequences are considered, especially if functional equivalence is not considered.

Although the numbers of genes that are most similar to *P. septica* are similar between the two *Mixta* species for both nucleotide and amino acid sequences (Figure 5), there were many genes in which the same species was the not the most similar taxa to both *Mixta* species. There were 86 genes for the nucleotide sequences and 147 genes for the amino acid sequences. However, of the 86 genes for nucleotide sequences, it was the case for 52% that one of the *Mixta* sequences was most similar to one representative of a genus and the other *Mixta* sequences was most similar to the other species of the same genus. This was also the case for 62% of the 147 amino acid genes.

Furthermore, percent identity analyses give only how similar are two sequences. They do not use evolutionary models or penalize transitions and transversions in nucleotide sequences. However, because mutations that do not disrupt any vital processes are rare, percent identity analyses may lend support to relatedness between two taxa.

### Spatial Patterns of Most Similar Taxa
Similar to what was done with the closest relatives in the genetic distance analysis, spatial patterns of the most similar taxa to the *M. calida* sequences based on site identity were investigated for the nucleotide sequences (Figure 6) and the amino acid sequences (Figure 7). Once again, no discernible patterns arose when spatial position on the *M. calida* genome are considered when either the nucleotide or amino acid sequences are analyzed (Figure 6 and 7). However, majority of the nucleotide genes that were most similar to the two *Tatumella* species and to *E. amylovora* occur in the first half of the genome (Figure 6).

```{r Figure6, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="Figure 6. The most similar taxa to *M. calida* genes for nucleotide sequences around the *M. calida* genome. The most similar taxa is defined to be that which sequence is most identical to the *M. calida* sequence. Length and colour of the bars represent the different species to which the gene is most similar.", fig.width=6.5, fig.height=5}
FI_calida_NT <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure6_CircularIdentities_NT.csv", 
                         stringsAsFactors = FALSE) %>%
  mutate(Closest_Relative = factor(Closest_Relative, 
                                   levels = c("P_septica", "P_agglomerans", "E_tasmaniensis", "E_amylovora", "T_ptyseos", 
                                              "T_saanichensis", "E_cloacae", "P_syringiae"), ordered = TRUE))

ggplot(FI_calida_NT, aes(x = ID, y = Relative_Number, fill = Closest_Relative)) +
  geom_bar(stat = "identity", position = position_dodge(), width = 15) +
  coord_polar() +
  scale_fill_brewer(palette = "Paired", 
                    labels = c("P. septica", "P. agglomerans", "E. tasmaniensis", "E. amylovora", "T. ptyseos", "T. saanichensis", "E. cloacae",
                               "P. syringae")) +
  scale_y_continuous(limits = c(0, 9), breaks = c(0, 2, 4, 6, 8)) +
  labs(fill = "Most Similar Taxa") +
  theme_bw() +
  theme(legend.text = element_text(face = "italic"),
        axis.title = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        text = element_text(size = 12,  family = "Times New Roman"))
```

```{r Figure7, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="Figure 7. The most similar taxa to *M. calida* genes for amino acid sequences around the *M. calida* genome. The most similar taxa is defined to be that which sequence is most identical to the *M. calida* sequence. Length and colour of the bars represent the different species to which the gene is most similar.", fig.width=6.5, fig.height=5}
FI_calida_AA <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure7_CircularIdentities_AA.csv", 
                         stringsAsFactors = FALSE) %>%
  mutate(Closest_Relative = factor(Closest_Relative, 
                                   levels = c("P_septica", "P_agglomerans", "E_tasmaniensis", "E_amylovora", "T_ptyseos", 
                                              "T_saanichensis", "E_cloacae", "P_syringiae"), ordered = TRUE))

ggplot(FI_calida_AA, aes(x = ID, y = Relative_Number, fill = Closest_Relative)) +
  geom_bar(stat = "identity", position = position_dodge(), width = 15) +
  coord_polar() +
  scale_fill_brewer(palette = "Paired", 
                    labels = c("P. septica", "P. agglomerans", "E. tasmaniensis", "E. amylovora", "T. ptyseos", "T. saanichensis", "E. cloacae",
                               "P. syringae")) +
  scale_y_continuous(limits = c(0, 9), breaks = c(0, 2, 4, 6, 8)) +
  labs(fill = "Most Similar Taxa") +
  theme_bw() +
  theme(legend.text = element_text(face = "italic"),
        axis.title = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        text = element_text(size = 12,  family = "Times New Roman"))
```

While no distinct patterns emerged, genes that are most similar to *P. septica* predominates the whole of the *M. calida* genome (Figures 6 and 7). This suggests that *Mixta* and *P. septica* may be related as their sequences are similar and few mutations have occurred after they diverged.

## A Comparison Between Evolutionary Change and Site Identity (Keep?)
### *Mixta calida*
Since all four of the analyses gave contentious results, it was necessary to compare the results for the first relative from genetic distance and the most similar taxa from site identity for both the nucleotide and amino acid sequences. For approximately 39% of the *M. calida* genes, all four analyses gave the same species. Of these genes, 255 agreed with *P. septica*, 22 with *P. agglomerans*, 17 with *E. tasmaniensis*, 8 with *E. amylovora*, 1 with *T. saanichensis*, and 6 with *E. cloacae*. Additionally, the four analyses gave the same genus approximately 63% of the genes where in 454 agreed with *Pantoea* followed by *Erwinia* with 44, and *Tatumella* with 3. Thus, because all four of the analyses agreed on a single species or genus, these results are more likely to be correct.

The two analyses conducted with nucleotide sequences agreed on a species 78% of the time whereas the two analyses conducted with amino acid sequences agreed only 70% of the time. It is not unexpected that the two nucleotide analyses would agree on taxa more often than the amino acid analyses since there are more sites in nucleotide sequences and no degeneracy in the code.

```{r Figure8, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="Figure 8. Sample for first 100 genes.", fig.width=6.5, fig.height=3}
all_results_calida <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/FigureN_AllResultsCalida.csv", 
                               stringsAsFactors = TRUE) %>%
  pivot_longer(cols = c(Rel_NT:Ide_AA), names_to = "Analysis", values_to = "Closest_Similar") %>%
  mutate(Closest_Similar = factor(Closest_Similar, 
                                  levels = c("P_septica", "P_agglomerans", "E_tasmaniensis", "E_amylovora", "T_ptyseos", "T_saanichensis", 
                                             "E_cloacae", "P_syringiae"), 
                                  ordered = TRUE),
         Analysis = factor(Analysis, levels = c("Ide_AA", "Rel_AA", "Ide_NT", "Rel_NT"), ordered = TRUE),
         Gene_Name = reorder(Gene_Name, ID))

ggplot(all_results_calida[1:400, ], aes(x = Gene_Name, y = Analysis, fill = Closest_Similar)) +
  geom_tile() +
  scale_fill_brewer(palette = "Paired", 
                    labels = c("P. septica", "P. agglomerans", "E. tasmaniensis", "E. amylovora", "T. ptyseos", "T. saanichensis", "E. cloacae",
                               "P. syringae")) +
  scale_y_discrete(labels = c("AA Identity", "AA Closest", "NT Identity", "NT Closest")) +
  labs(fill = "Closest Relative or \n Most Similar Taxa") +
  theme(legend.text = element_text(face = "italic"),
        # legend.position = "bottom",
        text = element_text(size = 12,  family = "Times New Roman"),
        axis.text.x = element_text(angle = 45, hjust = 1))               # Rotates x axis labels

stats_on_all <- read.csv("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/FigureO_AllStatsCalida.csv", stringsAsFactors = FALSE)

# table(stats_on_all$Same_Species)
# table(stats_on_all$Same_Genus)
# table(stats_on_all$NT_Same)
# table(stats_on_all$AA_Same)
# table(stats_on_all$Rel_Same)
# table(stats_on_all$Ide_Same)
```
Visual comparison of the results from the four analyses does not reveal any patterns (Figure 8). The four analyses are plotted on y axis where NT Closest is the closest relative according to genetic distance for nucleotide sequences  (Figure 8), NT Identity is the most similar taxa according to percent identity of nucleotide sequences, AA Closest is the closest relative according to genetic distance for amino acid sequences, and AA Identity is the most similar taxa according to percent identity of amino acid sequences (Figure 8). The first 100 genes according to *M. calida* gene ID are plotted on the x axis and the colour of the block represents the species that is the closest relative or most similar taxa to that gene for that analysis (Figure 8).

## Phylogenetic Trees
### Single Gene Trees
From the genetic distance analyses, only one gene separated *M. calida* and *M. gaviniae* for the nucleotide sequence and two genes for the amino acid sequence. The one gene for the nucleotide sequences was ttuB. According to genetic distance, *M. calida* was more closely related to *T. ptyseos* and *M. gaviniae* was more closely related to *P. septica* rather than to each other. A phylogenetic tree of this gene confirms that *M. calida* is more closely related to the *Tatumella* species where *M. gaviniae* to *Pantoea* and *Erwinia* (Figure 9A). The gene ttuB also had the two *Mixta* species separated for the amino acid sequences and the resulting tree shows the same results (Figure 9B).

```{r Figure9, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="Figure 9. Maximum likelihood gene trees for ttuB for nucleotide (A) and amino acid sequences (B).", fig.width=3.25}
knitr::include_graphics("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure9_ttuB_NT.png")
knitr::include_graphics("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure9_ttuB_AA.png")
```

The second gene that had the two *Mixta* species separated for the amino acid sequences was secY. From the nucleotide sequences, all of the taxa except for *P. syringae* are very closely related and bifurcations cannot be determined from the radial tree (Figure 10A). The same is the case for the amino acid sequences (Figure 10B). Condensed tree at 70% cutoff was created to show the relationships for both the nucleotide sequences (Figure 10C) and the amino acid sequences (Figure 10D). In the nucleotide tree, the two *Mixta* species are paired together, yet they are placed in a group with the two outgroups and the *Erwinia* species (Figure 10C). A condensed tree for the amino acid sequences showed that birfucations between most of the species were not supported above the 70% cutoff. These gene trees suggest that the differences between the *Enterobacterales* are not great enough to accurately infer the *Mixta* relative.

```{r Figure10, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="Figure 10. Maximum likelihood gene trees for secY for nucleotide (A) and amino acid sequences (B). Since bifucations could not be seen in radial trees, condensed trees were computed with a 70% cutoff for both the nucleotide sequnces (C) and amino acid sequences (D).", fig.width=3.25}
knitr::include_graphics("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure10_secY_NT.png")
knitr::include_graphics("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure10_secY_AA.png")
knitr::include_graphics("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure10b_secY_NT70.png")
knitr::include_graphics("C:/Users/Kim/OneDrive/2020_3Fall/Biology_396/10_Tables_and_Figures/Figure10b_secY_AA70.png")
```

# Conclusion
Since the analyses provided different results for many of the genes, a single analysis may not be enough to conclude that a species is related to another.

\newpage
# Literature Cited
<div id="refs"></div>

\newpage
# Appendix